---
title: "05-metabolomics-analysis"
author: "Yaamini Venkataraman"
date: '2025-11-11'
output: html_document
---

In this script, I will analyze metabolomics data from the West Coast Metabolomics Center (UC Davis). I will conduct an initial assessment of the data (PLS-DA), understand important metabolites and pathways (VIP analysis and WGCNA), and prepare for additional network analysis through MetaboAnalyst.

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../output/05-metabolomics-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, warning = FALSE, message = FALSE}
# if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse')
# if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan')
# if ("caret" %in% rownames(installed.packages()) == 'FALSE') install.packages('caret')
# if (!requireNamespace("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
# if ("mixOmics" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("mixOmics")
# if ("RVAideMemoire" %in% rownames(installed.packages()) == 'FALSE') install.packages('RVAideMemoire')
# if ("broom" %in% rownames(installed.packages()) == 'FALSE') install.packages('broom')
# if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer')
# if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap')
# if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("genefilter")
# if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('WGCNA')
# if ("dendsort" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('dendsort')
# if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('ComplexHeatmap')
# if ("svglite" %in% rownames(installed.packages()) == 'FALSE') install.packages("svglite")
# if ("ggfortify" %in% rownames(installed.packages()) == 'FALSE') install.packages("ggfortify")
# if (!requireNamespace("devtools", quietly = TRUE))
#     install.packages("devtools")
# devtools::install_github("andjar/ALASCA", ref = "main")
require(tidyverse)
require(vegan)
require(caret)
require(mixOmics)
require(RVAideMemoire)
require(broom)
require(RColorBrewer)
require(pheatmap)
require(genefilter)
require(WGCNA)
require(dendsort)
require(ComplexHeatmap)
require(svglite)
require(ggfortify)
require(devtools)
require(ALASCA)
```

```{r}
source("../../code/biostats.R") #Source multivariate analysis functions
```

```{r}
sessionInfo()
```

# Import data

```{r}
rawMetabolomicsData <- read.csv("../../data/raw-metabolomics-data-2023.csv", header = TRUE) %>% 
  dplyr::rename_with(~stringr::str_replace_all(., "X", "")) %>%
  mutate(BinBase.name = str_trim(BinBase.name, "both")) #Import data. Remove the "X" prefix for crab IDs. Remove white space from both sides of name
head(rawMetabolomicsData) #Confirm import. Crab IDs have "." instead of "-"
```

```{r}
metabolomicsMetadata <- read.csv("../../metadata/metabolomics-sample-list.csv", header = TRUE) %>%
  dplyr::select(1:3) %>%
  mutate(., day = case_when(date == "6/28/23" ~ "0",
                            date == "6/29/23" ~ "01",
                            date == "7/5/23" ~ "07",
                            date == "7/19/23" ~ "21",
                            date == "8/9/23" ~ "42") %>%
           as.character(.)) %>%
  dplyr::select(-date) %>%
  mutate(., treatment = case_when(treatment == "5C" ~ "05C",
                                  treatment != "5C" ~ treatment)) %>%
  mutate(., treatment_day = paste(treatment, day, sep = "_"))
#Import metadata. Select columns that actually have data
head(metabolomicsMetadata) #Confirm changes
```

```{r}
genotypeData <- read.csv("../../data/genotype/2023-crab-genotype.csv", header = TRUE, na.strings = "") %>%
  filter(., is.na(final.genotype) == FALSE) %>%
  dplyr::select(., crab.ID, treatment, final.genotype) %>%
  mutate(., presenceC = case_when(final.genotype == "CC" ~ "Y",
                                  final.genotype == "CT" ~ "Y",
                                  final.genotype == "TT" ~ "N"), 
         presenceT = case_when(final.genotype == "CC" ~ "N",
                               final.genotype == "CT" ~ "Y",
                               final.genotype == "TT" ~ "Y"), 
         het = case_when(final.genotype == "CC" ~ "N",
                         final.genotype == "CT" ~ "Y",
                         final.genotype == "TT" ~ "N")) #Import genotype data. Add additional informational columns
head(genotypeData) #Confirm genotype dataframe creation
```

# Format data

Before analyzing the data, I will transform the data and match with treatment metadata. The West Coast Metabolomics Center has already normalized data using the average mTIC (sum of all peak heights for *identified* metabolites) of the samples. 

REMOVE CRAB 5-064 SINCE IT DOESN'T HAVE AN RNA-SEQ PAIR.

```{r}
transMetabData <- metabolomicsMetadata %>%
  left_join(x = ., y = genotypeData, by = "crab.ID") %>%
  dplyr::rename(treatment = treatment.x,
                original.treatment = treatment.y) %>%
  mutate(., 
         treatment_genotype = paste(treatment, final.genotype, sep = "_"),
         treatment_presenceC = paste(treatment, presenceC, sep = "_"),
         treatment_presenceT = paste(treatment, presenceT, sep = "_"),
         treatment_het = paste(treatment, het, sep = "_")) %>%
  left_join(x = .,
            y = rawMetabolomicsData %>%
              dplyr::select(., -c(2:8)) %>%
              column_to_rownames(., var = "BinBase.name") %>%
              t(.) %>% as.data.frame(.) %>%
              rownames_to_column(., var = "crab.ID") %>% 
              mutate(crab.ID = gsub(x = crab.ID, pattern = "\\.", replacement = "-")),
            by = "crab.ID") %>%
  filter(., crab.ID != "5-064")
  #Remove columns with MS information, convert BinBase.name to rownames, and transform dataframe. Convert rownames to crab.ID column. Replace . with - in IDs. Join with genotype data and metadata. Since some samples were taken from 25 and 5 treatments before temperatures were changed (day 0), the "treatment" information used for metabolomics is slightly different than the original treatment assignment. Modify different treatment column names to reflect this. Remove sample 5-064 since it doesn't have an RNA-Seq pair.
head(transMetabData) #Confirm changes
nrow(transMetabData) #87 samples, which is how many were sequenced that have a RNA-Seq pair.
```

```{r}
transMetabData[,-c(1:13)] #All data
transMetabData[,c(14:198)] #Known data
```

# Analysis overview

I have three different questions I am trying to answer with the metabolomic data:

1. Is there temperature-specific differences in the metabolome?

> To answer this question, I would compare 15ºC vs. 5ºC and 25ºC, irrespective of time. This first question is more of a "sanity check" since I already know temperature has an outsized impact on the metabolome from previous experiments!

2. Are there different metabolic pathways activated over time in each temperature treatment?

> I can't answer this question by looking at all metabolomes, since my 15ºC control is not specific to any timepoint (control crabs were taken from throughout the experiment to use as a representative sample). Therefore, I just need to look at 25ºC and 5ºC for days 1 onwards.

3. Do different supergene genotypes have varied responses to chronic temperature exposure?

> This is where I can compare control crabs with the crabs sampled at day 42, taking into account the crab's genotype. Essentially, this would be a temperature x genotype interaction test, since the control is both day 0 and 15ºC.

# Impact of temperature on the metabolome

```{bash}
mkdir PCA
mkdir PCA/figures
```

```{r}
plotColors <- c(brewer.pal(9, "Reds")[7],
                brewer.pal(9, "Greys")[7],
                brewer.pal(9, "Blues")[7]) #Create color scheme
```

This is where I would compare 15ºC vs. 5ºC and 25ºC, irrespective of time.

## PCA

### All metabolites

```{r}
scaled.pca.T <- prcomp(transMetabData[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.T) #Standard deviations for each PC
```

```{r PCA interpretation}
pca.eigenval(scaled.pca.T) #Eigenvalues greater than the broken-stick expectation are considered to explain a statistically significance proportion of the variance in the original dataset. PC3 and PC4 eigenvalue > BSV
ordiMonteRandTest <- ordi.monte(transMetabData[-c(1:13)], ord = "pca", dim = 6, plot = FALSE) #Conduct monte carlo randomization test using 1000 random permutations of the input data.
ordiMonteRandTest

#All of these PCs are significant! Will move forward with PC1-2
```

```{r}
pcaPlot1 <- autoplot(scaled.pca.T, data = transMetabData, x = 1, y = 2,
                     size = 3, alpha = 0.8, colour = "treatment", shape = "treatment", 
                     frame = TRUE, frame.colour = "treatment", loadings = FALSE) +
  scale_color_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                     labels = c("5", "15", "25"),
                     name = "Temperature (ºC)") +
  scale_fill_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                    labels = c("5", "15", "25"),
                    name = "Temperature (ºC)") +
  scale_shape_manual(values = c(15, 19, 17),
                     labels = c("5", "15", "25"),
                     name = "Temperature (ºC)") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each treatment Do not include loadings. Manually define color, fill, and shapes.
pcaPlot1
ggsave("PCA/figures/all-data-PCA-treatment.pdf", height = 8.5, width = 11)
```

### Identified metabolites

I'm going to make a very similar PCA to the one above, but only using metabolites identified by the sequencer.

```{r}
transMetabData[c(14:198)] #All known metabolites
ncol(transMetabData[c(14:198)]) #There are 185 known metabolites
```

```{r}
scaled.pca.T.id <- prcomp(transMetabData[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.T.id) #Standard deviations for each PC
```

```{r}
pcaPlot2 <- autoplot(scaled.pca.T.id, data = transMetabData, x = 1, y = 2,
                     size = 3, alpha = 0.8, colour = "treatment", shape = "treatment", 
                     frame = TRUE, frame.colour = "treatment", loadings = FALSE) +
  scale_color_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                     labels = c("5", "15", "25"),
                     name = "Temperature (ºC)") +
  scale_fill_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                    labels = c("5", "15", "25"),
                    name = "Temperature (ºC)") +
  scale_shape_manual(values = c(15, 19, 17),
                     labels = c("5", "15", "25"),
                     name = "Temperature (ºC)") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot2
ggsave("PCA/figures/known-metabolites-PCA-treatment.pdf", height = 8.5, width = 11)
```

## Conduct PERMANOVA

### All metabolites

```{r}
dissim.T <- vegdist(scale(transMetabData[-c(1:13)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.T <- adonis2(dissim.T ~ as.factor(treatment), data = transMetabData, method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.T #Significant influence of temperature on metabolomes
```

```{r}
broom::tidy(permanova.T) %>%
  write.csv(., "PCA/all-data-PERMANOVA-treatment.csv", quote = FALSE) #Save PERMANOVA output
```

I will need to use a PERMDISP test to determine if significant differences are due to centroid or variance differences. I will do a separate test for each variable.

```{r}
disp.T <- betadisper(dissim.T, group = as.factor(transMetabData$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.T) #Dispersion differences exist
boxplot(disp.T) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.T) #Variance is different between groups. Temperature significance is due to centroid and dispersion differences (more dispersion in 15 and 25 than 5)
```

```{r}
broom::tidy(anova(disp.T)) %>%
  write.csv("PCA/all-data-PERMDISP-treatment.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

### Identified metabolites

```{r}
dissim.T.id <- vegdist(scale(transMetabData[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.T.id <- adonis2(dissim.T.id ~ as.factor(treatment), data = transMetabData, method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.T.id #Significant influence of temperature on metabolomes
```

```{r}
broom::tidy(permanova.T.id) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-treatment.csv", quote = FALSE) #Save PERMANOVA output
```

I will need to use a PERMDISP test to determine if significant differences are due to centroid or variance differences. I will do a separate test for each variable.

```{r}
disp.T.id <- betadisper(dissim.T.id, group = as.factor(transMetabData$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.T.id) #Dispersion differences exist
boxplot(disp.T.id) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.T.id) #Variance is different between groups. Temperature significance is due to centroid and dispersion differences (more dispersion in 15 and 25 than 5)
```

```{r}
broom::tidy(anova(disp.T.id)) %>%
  write.csv("PCA/known-metabolites-PERMDISP-treatment.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

The temperature impact on metabolomes is not as interesting of a question with this dataset, so I won't continue on to a PLS-DA. This does, however, give me necessary supplementary material to show an impact of temperature on metabolomes that needs to be considered for downstream analyses.

# Impact of time within temperature-specific metabolomes

Now that I have established that temperature has an impact on crab metabolomes, I want to know how how these metabolomes changed over time. I'll examine temperature x time interactions for the 25ºC and 5ºC data (no control data since it is not associated with a specific timepoint).

## PCA

### All metabolites

```{r}
(transMetabData %>% filter(., day != "0"))[-c(1:13)] #Data set without control crabs (day 0)
```

```{r}
scaled.pca.TT <- prcomp((transMetabData %>% filter(., day != "0"))[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TT) #Standard deviations for each PC
```

```{r}
pcaPlot3 <- autoplot(scaled.pca.TT, data = (transMetabData %>% filter(., day != "0")), x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "day", shape = "day",
                     frame = TRUE, frame.colour = "day", loadings = FALSE) +
  scale_color_manual(values = rainbow(4),
                     labels = c("1", "7", "21", "42"),
                     name = "Day") +
  scale_fill_manual(values = rainbow(4),
                    labels = c("1", "7", "21", "42"),
                    name = "Day") +
  scale_shape_manual(values = c(1, 2, 0, 4),
                     labels = c("1", "7", "21", "42"),
                     name = "Day") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Same as above but color by day and not treatment
pcaPlot3
ggsave("PCA/figures/all-data-PCA-day.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot4 <- autoplot(scaled.pca.TT, data = (transMetabData %>% filter(., day != "0")), x = 1, y = 2,
                     size = 4, alpha = 0.8, colour = "treatment_day", shape = "treatment_day", 
                     frame = TRUE, frame.colour = "treatment_day", loadings = FALSE) +
  scale_color_manual(values = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                                "salmon4", "salmon2", "firebrick1", plotColors[1]),
                     labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                                "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                     name = "Treatment") +
  scale_fill_manual(values = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                               "salmon4", "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                               "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                    name = "Treatment") +
  scale_shape_manual(values = c(1, 2, 0, 4, 1, 2, 0, 4),
                     labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                                "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                     name = "Treatment") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot4
ggsave("PCA/figures/all-data-PCA-temperatureDay.pdf", height = 8.5, width = 11)
```

### Identified metabolites

```{r}
(transMetabData %>% filter(., day != "0"))[c(14:198)] #Data set without control crabs (day 0)
```


```{r}
scaled.pca.TT.id <- prcomp((transMetabData %>% filter(., day != "0"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TT.id) #Standard deviations for each PC
```

```{r}
pcaPlot5 <- autoplot(scaled.pca.TT.id, data = (transMetabData %>% filter(., day != "0")), x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "day", shape = "day",
                     frame = TRUE, frame.colour = "day", loadings = FALSE) +
  scale_color_manual(values = rainbow(4),
                     labels = c("1", "7", "21", "42"),
                     name = "Day") +
  scale_fill_manual(values = rainbow(4),
                    labels = c("1", "7", "21", "42"),
                    name = "Day") +
  scale_shape_manual(values = c(1, 2, 0, 4),
                     labels = c("1", "7", "21", "42"),
                     name = "Day") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Same as above but color by day and not treatment
pcaPlot5
ggsave("PCA/figures/known-metabolites-PCA-day.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot6 <- autoplot(scaled.pca.TT.id, data = (transMetabData %>% filter(., day != "0")), x = 1, y = 2,
                     size = 4, alpha = 0.8, colour = "treatment_day", shape = "treatment_day", 
                     frame = TRUE, frame.colour = "treatment_day", loadings = FALSE) +
  scale_color_manual(values = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                                "salmon4", "salmon2", "firebrick1", plotColors[1]),
                     labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                                "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                     name = "Treatment") +
  scale_fill_manual(values = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                               "salmon4", "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                               "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                    name = "Treatment") +
  scale_shape_manual(values = c(1, 2, 0, 4, 1, 2, 0, 4),
                     labels = c("5ºC at Day 1", "5ºC at Day 7", "5ºC at Day 21", "5ºC at Day 42",
                                "25ºC at Day 1", "25ºC at Day 7", "25ºC at Day 21", "25ºC at Day 42"),
                     name = "Treatment") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot6
ggsave("PCA/figures/known-metabolites-PCA-temperatureDay.pdf", height = 8.5, width = 11)
```

## Conduct PERMANOVA

### All metabolites

```{r}
(transMetabData %>% filter(., day != "0"))[-c(1:13)]
```

```{r}
dissim.TT <- vegdist(scale((transMetabData %>% filter(., day != "0"))[-c(1:13)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.TT <- adonis2(dissim.TT ~ as.factor(treatment)*as.factor(day), data = (transMetabData %>% filter(., day != "0")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TT #Significant influence of temperature, day, and their interaction on metabolomes
```

```{r}
broom::tidy(permanova.TT) %>%
  write.csv(., "PCA/all-data-PERMANOVA-temperatureDay.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
disp.TT.temp <- betadisper(dissim.TT, group = as.factor((transMetabData %>% filter(., day != "0"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.temp) #Dispersion differences exist
boxplot(disp.TT.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.temp) #Variance is different between groups. Temperature significance is due to centroid and dispersion differences
```

```{r}
disp.TT.day <- betadisper(dissim.TT, group = as.factor((transMetabData %>% filter(., day != "0"))$day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.day) #Dispersion differences exist
boxplot(disp.TT.day) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.day) #Variance is similar between groups. Day significance is due to centroid differences only
```

```{r}
disp.TT.tempDay <- betadisper(dissim.TT, group = as.factor((transMetabData %>% filter(., day != "0"))$treatment_day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.tempDay) #Dispersion differences exist
boxplot(disp.TT.tempDay) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.tempDay) #Variance is not different between groups. Interaction significance is due to centroid differences only
```

```{r}
rbind(broom::tidy(anova(disp.TT.temp)) %>%
        mutate(var = "temperature"),
      broom::tidy(anova(disp.TT.day)) %>%
        mutate(var = "day"),
      broom::tidy(anova(disp.TT.tempDay)) %>%
        mutate(var = "interaction")) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/all-data-PERMDISP-temperatureDay.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

### Identified metabolites

```{r}
(transMetabData %>% filter(., day != "0"))[c(14:198)]
```

```{r}
dissim.TT.id <- vegdist(scale((transMetabData %>% filter(., day != "0"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.TT.id <- adonis2(dissim.TT.id ~ as.factor(treatment)*as.factor(day), data = (transMetabData %>% filter(., day != "0")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TT.id #Significant influence of temperature, day, and their interaction on metabolomes
```

```{r}
broom::tidy(permanova.TT.id) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-temperatureDay.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
disp.TT.id.temp <- betadisper(dissim.TT.id, group = as.factor((transMetabData %>% filter(., day != "0"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.id.temp) #Dispersion differences exist
boxplot(disp.TT.id.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.id.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TT.id.day <- betadisper(dissim.TT.id, group = as.factor((transMetabData %>% filter(., day != "0"))$day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.id.day) #Dispersion differences exist
boxplot(disp.TT.id.day) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.id.day) #Variance is similar between groups. Day significance is due to centroid differences only
```

```{r}
disp.TT.id.tempDay <- betadisper(dissim.TT.id, group = as.factor((transMetabData %>% filter(., day != "0"))$treatment_day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TT.id.tempDay) #Dispersion differences exist
boxplot(disp.TT.id.tempDay) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TT.id.tempDay) #Variance is not different between groups. Interaction significance is due to centroid differences only
```

```{r}
rbind(broom::tidy(anova(disp.TT.id.temp)) %>%
        mutate(var = "temperature"),
      broom::tidy(anova(disp.TT.id.day)) %>%
        mutate(var = "day"),
      broom::tidy(anova(disp.TT.id.tempDay)) %>%
        mutate(var = "interaction")) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-temperatureDay.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

## ASCA

I will use an ASCA to decompose the trends I observed in the PERMANOVA into groups of metabolites with similar abundance patterns. In conjunction with a PLS-DA, I can use the information to determine the best pairwise comparisons to use to pull out metabolites of interest.

```{bash}
mkdir ASCA
mkdir ASCA/figures
```

### Format data

```{r}
tempTime.ASCA <- transMetabData[c(1:198)] %>%
  filter(., day != "0") %>%
  dplyr::select(-c(treatment:treatment_het)) %>%
  pivot_longer(., cols = !crab.ID, names_to = "variable", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  left_join(., metabolomicsMetadata, by = "crab.ID") %>%
  left_join(., genotypeData, by = "crab.ID") %>%
  dplyr::rename(treatment = treatment.x,
                original.treatment = treatment.y) %>%
  dplyr::select(., c(crab.ID:day, final.genotype)) %>%
  mutate(., treatment = as.factor(treatment),
         day = as.factor(day))
#Take known metabolomics data and filter day 0 information. Rmeove extraneous columns. Pivot data longer for ASCA. Compound information should be "variable." Add metadata to dataframe with all compound information and retain only useful columns (treatment, day, and final genotype). Convert variables to factors.
head(tempTime.ASCA) #Confirm changes
```

### Run model

```{r}
tempTime.ASCAmodel <- ALASCA(df = tempTime.ASCA,
                             formula = value ~ day*treatment + (1|crab.ID),
                             n_validation_runs = 100,
                             validate = TRUE,
                             reduce_dimensions = TRUE) #Run ASCA to identify day and temperature effects on compound abundance. Include individual crab and genotype as random effects
```

### Interpret ASCA

```{r}
# pdf("ASCA/figures/tempTime-scree-plot.pdf")
plot(tempTime.ASCAmodel, component = 1:10, type = "scree") #Scree plot of variance explained by each PC. There were 7 components identified, with the eigth-tenth components explaining very little variance. There's a pretty steep drop-off between components 2 and 3.
# dev.off()
```

#### Component 1

This component shows metabolites with consistently higher abundance at 25ºC that reduce in abundance over time.

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 1, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-effect-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 1, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-prediction-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
tempTime.loadingsPC1 <- get_loadings(tempTime.ASCAmodel, component = 1, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempTime.loadingsPC1)
```

```{r}
tempTime.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempTime-loadings-PC1.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 2

This component shows metabolites with consistently higher abundance at 25ºC that increase in abundance over time.

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 2, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-effect-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 2, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-prediction-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
tempTime.loadingsPC2 <- get_loadings(tempTime.ASCAmodel, component = 2, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempTime.loadingsPC2)
```

```{r}
tempTime.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempTime-loadings-PC2.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 3

This component shows metabolites that have higher abundance at one temperature earlier on in the experiment, but then have higher abundance at the other temperature later in the experiment.

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 3, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-effect-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempTime.ASCAmodel, effect = 1, component = 3, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTime-prediction-plot-PC3.pdf", height = 11, width = 8.5)
```

Component 3 explains < 10% of the variance in this dataset. I will use a cutoff of 10% to select components for downstream analysis. This also matches up with the scree plot, where after 2 components adding an extra component doesn't substantially increase the total amount of variation explained.

## PLS-DA

I am going to use a PLS-DA to investigate significant effects identified with the PERMANOVA (temperature, time, and their interaction) using the `mixOmics` package. This is a more robust way to analyze differences between treatments and is common in many metabolomics studies. I am going to look at effects that were significant in identified metabolites, not all metabolites detected.

```{bash}
mkdir PLSDA
mkdir PLSDA/figures
ls #Confirm that directory was created in the correct location
```

### Format data

```{r}
X_tempTime <- transMetabData %>% 
  filter(., day != "0") %>%
  dplyr::select(c(14:198)) %>%
  log(.) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) #Assign known metabolite data to the X dataframe after log transformation. Replace any -Inf with NA, then replace NA with 0
Y_tempTime <- as.factor((transMetabData %>% filter(., day != "0"))$treatment_day) #Assign treatment data to the Y dataframe
```

```{r}
nzv_indices <- nearZeroVar(X_tempTime) #Check to see if any variables have near-zero variance.
nzv_indices$Position #There aren't any columns with near-zero variance!
```

### Tune PLS-DA model

First, I need to determine how many components to include in the analysis. I'll create an initial PCA, then run a permutation test to determine how many components to include.

```{r}
treatmentday.known.plsda <- plsda(X_tempTime, Y_tempTime, ncomp = 10) #Initially set a higher number for the number of components since we will be testing this
plotIndiv(treatmentday.known.plsda) #Preliminary PLS-DAxa
```

```{r}
perf.treatmentday.known.plsda <- perf(treatmentday.known.plsda, validation = "Mfold", folds = 5,  
                                      progressBar = TRUE, nrepeat = 50) 

plot(perf.treatmentday.known.plsda, sd = TRUE) #Overall, increasing the number of components decreases the error rate. There doesn't seem to be an exponential reduction of error rate after 3 components. Mahalanobis distance offers the lowest overall and balanced error rates (BER) when using 3 components
perf.treatmentday.known.plsda$choice.ncomp #Confirm information from plot.
```

### Run PLS-DA

```{r}
treatmentDay.known.plsda.tuned <- plsda(X_tempTime, Y_tempTime, ncomp = 3) #Run PLS-DA with the appropriate number of components to include based on tuning above.
plotIndiv(treatmentDay.known.plsda.tuned, comp = c(1,2)) #Preliminary PLS-DA
plotIndiv(treatmentDay.known.plsda.tuned, comp = c(1,3)) #Preliminary PLS-DA
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-treatmentDay-comp12.pdf", height = 8.5, width = 11)
plotIndiv(treatmentDay.known.plsda.tuned, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                  "salmon4", "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(rep(c(1, 2, 0, 4), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 19% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 11% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Do not include a legend. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-treatmentDay-comp13.pdf", height = 8.5, width = 11)
plotIndiv(treatmentDay.known.plsda.tuned, comp = c(1,3), ind.names = FALSE,
          col = c("lightblue4", "lightblue2", "steelblue1", plotColors[3], 
                  "salmon4", "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(rep(c(1, 2, 0, 4), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 19% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 9% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

### Validate PLS-DA

The last step is to know how many variables (metabolites) make sense to include with the optimal number of components. I can then run this reduced PLSDA to ensure the trends remain consistent.

```{r}
list.keepX <- c(5:10,  seq(20, 300, 10)) #list.keepX # to output the grid of values tested
```

```{r}
tune.splsda.treatmentDay.known <- tune.splsda(X_tempTime, Y_tempTime, ncomp = 3, 
                                              validation = 'Mfold', folds = 5,
                                              measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.treatmentDay.known) #Plot results
```

```{r}
error <- tune.splsda.treatmentDay.known$error.rate #Save error rate
ncomp <- tune.splsda.treatmentDay.known$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #3 components! That's in-line with the perf analysis and what I ended up using.

select.keepX <- tune.splsda.treatmentDay.known$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with this many features per component.
```

```{r}
final.splsda.treatmentDay.known <- splsda(X_tempTime, Y_tempTime, ncomp = ncomp, keepX = select.keepX) #Use the number of components and metabolites to keep based on optimization from above
plotIndiv(final.splsda.treatmentDay.known, comp = c(1,2))
plotIndiv(final.splsda.treatmentDay.known, comp = c(1,3))
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-treatmentDay-comp12.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.treatmentDay.known, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue4", "lightblue2", "steelblue1", plotColors[3],
                  "salmon4", "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(rep(c(1, 2, 0, 4), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by Temperature and Day", X.label = "Component 1: 18% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 11% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-treatmentDay-comp13.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.treatmentDay.known, comp = c(1,3), ind.names = FALSE,
          col = c("lightblue4", "lightblue2", "steelblue1", plotColors[3],
                  "salmon4", "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(rep(c(1, 2, 0, 4), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by Temperature and Day", X.label = "Component 1: 18% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 7% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

# Early vs. late comparisons

Based on the PCA, ASCA, and PLS-DAs above, it seems like there is a clear separation between early (days 1 and 7) and late (days 21 and 42) timepoints. I'm going to repeat my analyses using these larger time classifications for the known metabolites.

## Format data

```{r}
transMetabData_knownTimepoint <- transMetabData[c(1:198)] %>%
  filter(., day != "0") %>%
  mutate(., timepoint = case_when(day == "01" | day == "07" ~ "early",
                                  day == "21" | day == "42" ~ "late")) %>%
  mutate(., treatment_timepoint = paste(treatment, timepoint, sep = "_")) #Take known metabolites and remove day 0 samples. Create columns for timepoint (early vs late) and a column with combined treatment and timepoint information
head(transMetabData_knownTimepoint) #Confirm dataframe creation. Columns 14:198 would still be the metabolite data.
```

```{r}
transMetabData_knownTimepoint %>%
  group_by(., treatment, timepoint, final.genotype) %>%
  summarize(count = n())
```

I don't have enough crabs between all the genotypes to look at the impact of genotype along with temperature and timepoint, so I'll stick to analyzing genotype and temperature separately.

## PCA

```{r}
scaled.pca.TTP.id <- prcomp((transMetabData_knownTimepoint)[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TTP.id) #Standard deviations for each PC
```

```{r}
pcaPlot34 <- autoplot(scaled.pca.TTP.id, data = (transMetabData_knownTimepoint), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_timepoint", shape = "treatment_timepoint", 
                      frame = TRUE, frame.colour = "treatment_timepoint", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3], 
                                "salmon2", plotColors[1]),
                     labels = c("Early 5ºC", "Late 5ºC",
                                "Early 25ºC", "Late 25ºC"),
                     name = "Treatment") +
  scale_fill_manual(values = c("lightblue2", plotColors[3], 
                               "salmon2", plotColors[1]),
                    labels = c("Early 5ºC", "Late 5ºC",
                               "Early 25ºC", "Late 25ºC"),
                    name = "Treatment") +
  scale_shape_manual(values = c(1, 2, 1, 2),
                     labels = c("Early 5ºC", "Late 5ºC",
                                "Early 25ºC", "Late 25ºC"),
                     name = "Treatment") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot34
ggsave("PCA/figures/known-metabolites-PCA-temperatureTimepoint.pdf", height = 8.5, width = 11)
```

## Conduct PERMANOVA

```{r}
dissim.TTP.id <- vegdist(scale(transMetabData_knownTimepoint[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.TTP.id <- adonis2(dissim.TTP.id ~ as.factor(treatment)*as.factor(timepoint), data = (transMetabData_knownTimepoint), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TTP.id #Significant influence of temperature, day, and their interaction on metabolomes
```

```{r}
broom::tidy(permanova.TTP.id) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-temperaturTimepoint.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
disp.TTP.id.temp <- betadisper(dissim.TTP.id, group = as.factor((transMetabData_knownTimepoint)$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TTP.id.temp) #Plot PCoA
boxplot(disp.TTP.id.temp) #Show distance to group centroids
anova(disp.TTP.id.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TTP.id.timepoint <- betadisper(dissim.TTP.id, group = as.factor((transMetabData_knownTimepoint)$timepoint), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TTP.id.timepoint) #Plot PCoA
boxplot(disp.TTP.id.timepoint) #Show distance to group centroids
anova(disp.TTP.id.timepoint) #Variance is similar between groups. Timepoint significance is due to centroid differences only
```

```{r}
disp.TTP.id.tempTimepoint <- betadisper(dissim.TTP.id, group = as.factor((transMetabData_knownTimepoint)$treatment_timepoint), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TTP.id.tempTimepoint) #Dispersion differences exist
boxplot(disp.TTP.id.tempTimepoint) #Show distance to group centroids
anova(disp.TTP.id.tempTimepoint) #Variance is not different between groups. Interaction significance is due to centroid differences only
```

```{r}
rbind(broom::tidy(anova(disp.TTP.id.temp)) %>%
        mutate(var = "temperature"),
      broom::tidy(anova(disp.TTP.id.timepoint)) %>%
        mutate(var = "timepoint"),
      broom::tidy(anova(disp.TTP.id.tempTimepoint)) %>%
        mutate(var = "interaction")) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-temperatureTimepoint.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

## ASCA

### Format data

```{r}
tempTimepoint.ASCA <- transMetabData_knownTimepoint %>%
  dplyr::select(-c(treatment:treatment_het, timepoint:treatment_timepoint)) %>%
  pivot_longer(., cols = !crab.ID, names_to = "variable", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  left_join(., metabolomicsMetadata, by = "crab.ID") %>%
  left_join(., genotypeData, by = "crab.ID") %>%
  dplyr::rename(treatment = treatment.x,
                original.treatment = treatment.y) %>%
  mutate(., timepoint = case_when(day == "01" | day == "07" ~ "early",
                                  day == "21" | day == "42" ~ "late")) %>%
  dplyr::select(., c(crab.ID:treatment, timepoint)) %>%
  mutate(., treatment = as.factor(treatment),
         timepoint = as.factor(timepoint))
#Take known metabolomics data and filter day 0 information. Rmeove extraneous columns. Pivot data longer for ASCA. Compound information should be "variable." Add metadata to dataframe with all compound information and retain only useful columns (treatment, day, and final genotype). Convert variables to factors.
head(tempTimepoint.ASCA) #Confirm changes
```

### Run model

```{r}
tempTimepoint.ASCAmodel <- ALASCA(df = tempTimepoint.ASCA,
                                  formula = value ~ timepoint*treatment + (1|crab.ID),
                                  n_validation_runs = 100,
                                  validate = TRUE,
                                  reduce_dimensions = TRUE) #Run ASCA to identify day and temperature effects on compound abundance. Include individual crab and genotype as random effects
```

### Interpret ASCA

```{r}
# pdf("ASCA/figures/tempTimepoint-scree-plot.pdf")
plot(tempTimepoint.ASCAmodel, component = 1:10, type = "scree") #Scree plot of variance explained by each PC. There's a pretty steep drop-off between components 2 and 3. Component 3 may or may not explain more than 10% of the variance.
# dev.off()
```

#### Component 1

This component shows metabolites with consistently higher abundance at 25ºC that decrease in abundance over time.

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 1, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-effect-plot-PC1.pdf", height = 8.5, width = 11)
```

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 1, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-prediction-plot-PC1.pdf", height = 8.5, width = 11)
```

```{r}
tempTimepoint.loadingsPC1 <- get_loadings(tempTimepoint.ASCAmodel, component = 1, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempTimepoint.loadingsPC1)
```

```{r}
tempTimepoint.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempTimepoint-loadings-PC1.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 2

This component shows metabolites with consistently higher abundance at 25ºC that increase in abundance over time.

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 2, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-effect-plot-PC2.pdf", height = 8.5, width = 11)
```

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 2, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-prediction-plot-PC2.pdf", height = 8.5, width = 11)
```

```{r}
tempTimepoint.loadingsPC2 <- get_loadings(tempTimepoint.ASCAmodel, component = 2, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempTimepoint.loadingsPC2)
```

```{r}
tempTimepoint.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempTimepoint-loadings-PC2.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 3

This component shows metabolites that decrease in abundance over time at 25ºC but increase in abundance over time at 5ºC

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 3, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-effect-plot-PC3.pdf", height = 8.5, width = 11)
```

```{r}
plot(tempTimepoint.ASCAmodel, effect = 1, component = 3, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Timepoint",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempTimepoint-prediction-plot-PC3.pdf", height = 8.5, width = 11)
```

```{r}
tempTimepoint.loadingsPC3 <- get_loadings(tempTimepoint.ASCAmodel, component = 3, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempTimepoint.loadingsPC3)
```

```{r}
tempTimepoint.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempTimepoint-loadings-PC3.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

### Metaboanalyst

I will use Metaboanalyst to identify enriched pathways in for metabolites of interest. These metabolites of interest will come from ASCA components as well as significant VIP (identified later after doing a PLS-DA). To identify metabolites of interest for the ASCA, I will select metabolites that explain a certain amount of cumulative variance.

Once I have a list of metabolites, I will do the following within MetaboAnalyst:

- Uploaded each list of metabolite compound names for each ASCA component in the Pathway Analysis module
- Match compound names between the sequencer and Metaboanalyst as needed, based on HMDB and KEGG identifiers
- Matched to KEGG pathways from *Drosophila melanogaster*
- Review pathway analysis output and download for further review and visualization

```{bash}
mkdir metaboanalyst
mkdir metaboanalyst/figures
```

#### Component 1

```{r}
tempTimepoint.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempTimepoint.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("ASCA/tempTimepoint-PC1_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempTime_PC1_pathwayResults <- read.csv("metaboanalyst/tempTime-loadings-PC1_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempTime_PC1_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempTime_PC1_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempTime_PC1-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_tempTime_PC1_pathwayResults <- tempTime_PC1_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 pathways
head(top_tempTime_PC1_pathwayResults) #Confirm formatting
```

```{r}
top_tempTime_PC1_pathwayResults %>%
  mutate(., order = rev(1:24)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_tempTime_PC1_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempTime_PC1-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Starch and sucrose metabolism was significant for metabolites with higher abundance at 25ºC that increase over time. I will need to look at abundance of compounds in this pathway across temperature and timepoint.

Compounds in this pathway:

C00103: glucose-1-phosphate
C00092: glucose-6-phosphate
C00095: fructose
C00085: fructose-6-phosphate

```{r}
tempTimePC1_starchSucrose_facets <- c("1" = "glucose-1-phosphate",
                          "2" = "glucose-6-phosphate",
                          "3" = "fructose",
                          "4" = "fructose-6-phosphate") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     `glucose-1-phosphate`, `glucose-6-phosphate`, fructose, `fructose-6-phosphate`)) %>%
  pivot_longer(., cols = c(3:6), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "glucose-1-phosphate" ~ 1,
                              metabolite == "glucose-6-phosphate" ~ 2,
                              metabolite == "fructose" ~ 3,
                              metabolite == "fructose-6-phosphate" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = tempTimePC1_starchSucrose_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", plotColors[3], 
                               "salmon2", plotColors[1]),
                    labels = c("Early 5ºC", "Late 5ºC", 
                               "Early 25ºC", "Late 25ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", plotColors[3], 
                               "salmon2", plotColors[1]),
                    labels = c("Early 5ºC", "Late 5ºC", 
                               "Early 25ºC", "Late 25ºC")) +
  xlab("") + #Axis titles
  ggtitle("Starch and Sucrose Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/tempTime-PC1_starch-surcose-metabolism-plot.pdf", height = 8.5, width = 11)
```

#### Component 2

```{r}
tempTimepoint.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempTimepoint.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("ASCA/tempTimepoint-PC2_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempTime_PC2_pathwayResults <- read.csv("metaboanalyst/tempTime-loadings-PC2_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempTime_PC2_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempTime_PC2_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempTime_PC2-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

No significant pathways, will not proceed with further visualization.

#### Component 3

```{r}
tempTimepoint.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempTimepoint.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("ASCA/tempTimepoint-PC3_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempTime_PC3_pathwayResults <- read.csv("metaboanalyst/tempTime-loadings-PC3_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempTime_PC3_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempTime_PC3_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempTime_PC3-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

No significant pathways, will not proceed with further visualization.

## PLS-DA

### Format data

```{r}
X_tempTimepoint <- transMetabData_knownTimepoint %>% 
  dplyr::select(c(14:198)) %>%
  log(.) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) #Assign known metabolite data to the X dataframe after log transformation. Replace any -Inf with NA, then replace NA with 0
Y_tempTimepoint <- as.factor((transMetabData_knownTimepoint)$treatment_timepoint) #Assign treatment data to the Y dataframe
```

```{r}
nzv_indices <- nearZeroVar(X_tempTimepoint) #Check to see if any variables have near-zero variance.
nzv_indices$Position #There aren't any columns with near-zero variance!
```

### Tune PLS-DA model

First, I need to determine how many components to include in the analysis. I'll create an initial PCA, then run a permutation test to determine how many components to include.

```{r}
treatmentTimepoint.known.plsda <- plsda(X_tempTimepoint, Y_tempTimepoint, ncomp = 10) #Initially set a higher number for the number of components since we will be testing this
plotIndiv(treatmentTimepoint.known.plsda) #Preliminary PLS-DAxa
```

```{r}
perf.treatmentTimepoint.known.plsda <- perf(treatmentTimepoint.known.plsda, validation = "Mfold", folds = 5,  
                                            progressBar = TRUE, nrepeat = 50) 

plot(perf.treatmentTimepoint.known.plsda, sd = TRUE) #Overall, increasing the number of components decreases the error rate. There doesn't seem to be an exponential reduction of error rate after 5 components.
perf.treatmentTimepoint.known.plsda$choice.ncomp #Confirm information from plot.
```

### Run PLS-DA

```{r}
treatmentTimepoint.known.plsda.tuned <- plsda(X_tempTimepoint, Y_tempTimepoint, ncomp = 5) #Run PLS-DA with the appropriate number of components to include based on tuning above.
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,2)) #Preliminary PLS-DA: PC 1 = 19, PC 2 = 11
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,3)) #Preliminary PLS-DA: PC 3 = 9
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,4)) #Preliminary PLS-DA: PC 4 = 4
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,5)) #Preliminary PLS-DA: PC 5 = 5

#PCs 4 and 5 don't explain a lot of variation, so I'll stick to looking at components 1-3.
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-treatmentTimepoint-comp12.pdf", height = 8.5, width = 11)
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", plotColors[3], 
                  "salmon2", plotColors[1]),
          pch = as.factor(rep(c(1, 2), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by Temperature and Timepoint", X.label = "Component 1: 19% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 11% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Do not include a legend. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-treatmentTimepoint-comp13.pdf", height = 8.5, width = 11)
plotIndiv(treatmentTimepoint.known.plsda.tuned, comp = c(1,3), ind.names = FALSE,
          col = c("lightblue2", plotColors[3], 
                  "salmon2", plotColors[1]),
          pch = as.factor(rep(c(1, 2), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by Temperature and Timepoint", X.label = "Component 1: 19% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 9% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

### Validate PLS-DA

```{r}
tune.splsda.treatmentTimepoint.known <- tune.splsda(X_tempTimepoint, Y_tempTimepoint, ncomp = 5, 
                                                    validation = 'Mfold', folds = 5,
                                                    measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.treatmentTimepoint.known) #Plot results
```

```{r}
error <- tune.splsda.treatmentTimepoint.known$error.rate #Save error rate
ncomp <- tune.splsda.treatmentTimepoint.known$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #3 components! That's in-line with what I observed.

select.keepX <- tune.splsda.treatmentTimepoint.known$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with this many features per component.
```

```{r}
final.splsda.treatmentTimepoint.known <- splsda(X_tempTimepoint, Y_tempTimepoint, ncomp = ncomp, keepX = select.keepX) #Use the number of components and metabolites to keep based on optimization from above
plotIndiv(final.splsda.treatmentTimepoint.known, comp = c(1,2)) #PC1: 17, PC2: 12
plotIndiv(final.splsda.treatmentTimepoint.known, comp = c(1,3)) #PC3: 7%
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-treatmentTimepoint-comp12.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.treatmentTimepoint.known, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", plotColors[3],
                  "salmon2", plotColors[1]),
          pch = as.factor(rep(c(1, 2), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by Temperature and Timepoint", X.label = "Component 1: 17% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 12% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

```{r}
# pdf("PLSDA/figures/known-metabolites-sPLSDA-treatmentTimepoint-comp13.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.treatmentTimepoint.known, comp = c(1,3), ind.names = FALSE,
          col = c("lightblue2", plotColors[3],
                  "salmon2", plotColors[1]),
          pch = as.factor(rep(c(1, 2), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by Temperature and Timepoint", X.label = "Component 1: 17% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 7% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
# dev.off()
```

### Pairwise tests for VIP

```{r}
transMetabData_knownTimepoint$treatment_timepoint %>% as.factor() %>% levels() #List of descriptors for different treatments
```

#### Early 5ºC vs. Early 25ºC

```{r}
list <- c("05C_early", "25C_early") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_timepoint) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_5v25_early <- plsda(X, Y, ncomp = 3) #Run PLSDA for desired contrast
VIP_5v25_early <- as.data.frame(PLSDA.VIP(known.plsda_5v25_early)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_5v25_early
```

```{r}
clean_5v25_early <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_5v25_early <- cbind(clean_5v25_early[, c(1:2, 199:200)], 
                               log(clean_5v25_early[, names(clean_5v25_early) %in% VIP_5v25_early$metabolite])) %>%
  pivot_longer(., cols = 5:87, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_5v25_early #Confirm changes
```

```{r}
t.test_5v25_early <-do(VIP_select_5v25_early, tidy(t.test(.$log_norm_counts ~ .$treatment_timepoint,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_5v25_early$p.adj <- p.adjust(t.test_5v25_early$p.value, method = c("fdr"), n = length(VIP_5v25_early$metabolite)) #Adjust p value for the number of comparisons
t.test_5v25_early
```

```{r}
t.test_5v25_early %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #69 significantly different metabolites
t.test_5v25_early %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #1 metabolites higher in 5ºC
t.test_5v25_early %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #68 metabolites higher in 25ºC
```

```{r}
t.test_5v25_early %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  mutate(., higherAbundanceIn = case_when(estimate1 > estimate2 ~ "05C_early",
                                          estimate1 < estimate2 ~ "25C_early")) %>%
  dplyr::select(., metabolite, PubChem, KEGG, higherAbundanceIn) %>%
  unique(.) %>%
  write.table("PLSDA/5v25-early_significantVIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with identifiers. Create a column with information on condition with the higher abundance. Retain unique entries and write table.
```

```{r}
t.test_5v25_early %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_5v25_early, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "5",
                                  estimate < 0 ~ "25")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[3], plotColors[1]), 
                    name = "Temperature (ºC)",
                    breaks = c("5", "25"),
                    labels = c("5", "25")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-5v25-early.pdf", height = 8.5, width = 11)
```

#### Late 5ºC vs. Late 25ºC

```{r}
list <- c("05C_late", "25C_late") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_timepoint) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_5v25_late <- plsda(X, Y, ncomp = 3) #Run PLSDA for desired contrast
VIP_5v25_late <- as.data.frame(PLSDA.VIP(known.plsda_5v25_late)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_5v25_late
```

```{r}
clean_5v25_late <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_5v25_late <- cbind(clean_5v25_late[, c(1:2, 199:200)], 
                              log(clean_5v25_late[, names(clean_5v25_late) %in% VIP_5v25_late$metabolite])) %>%
  pivot_longer(., cols = 5:66, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_5v25_late #Confirm changes
```

```{r}
t.test_5v25_late <-do(VIP_select_5v25_late, tidy(t.test(.$log_norm_counts ~ .$treatment_timepoint,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_5v25_late$p.adj <- p.adjust(t.test_5v25_late$p.value, method = c("fdr"), n = length(VIP_5v25_late$metabolite)) #Adjust p value for the number of comparisons
t.test_5v25_late
```

```{r}
t.test_5v25_late %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  mutate(., higherAbundanceIn = case_when(estimate1 > estimate2 ~ "05C_late",
                                          estimate1 < estimate2 ~ "25C_late")) %>%
  dplyr::select(., metabolite, PubChem, KEGG, higherAbundanceIn) %>%
  unique(.) %>%
  write.table("PLSDA/5v25-late_significantVIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with identifiers. Create a column with information on condition with the higher abundance. Retain unique entries and write table.
```

```{r}
t.test_5v25_late %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #60 significantly different metabolites
t.test_5v25_late %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #14 metabolites higher in 5ºC
t.test_5v25_late %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #46 metabolites higher in 25ºC
```

```{r}
t.test_5v25_late %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_5v25_late, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "5",
                                  estimate < 0 ~ "25")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[3], plotColors[1]), 
                    name = "Temperature (ºC)",
                    breaks = c("5", "25"),
                    labels = c("5", "25")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-5v25-late.pdf", height = 8.5, width = 11)
```

#### Early 5ºC vs. Late 5ºC

```{r}
list <- c("05C_early", "05C_late") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_timepoint) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_5_earlyvlate <- plsda(X, Y, ncomp = 3) #Run PLSDA for desired contrast
VIP_5_earlyvlate <- as.data.frame(PLSDA.VIP(known.plsda_5_earlyvlate)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_5_earlyvlate
```

```{r}
clean_5_earlyvlate <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_5_earlyvlate <- cbind(clean_5_earlyvlate[, c(1:2, 199:200)], 
                                 log(clean_5_earlyvlate[, names(clean_5_earlyvlate) %in% VIP_5_earlyvlate$metabolite])) %>%
  pivot_longer(., cols = 5:56, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_5_earlyvlate #Confirm changes
```

```{r}
t.test_5_earlyvlate <-do(VIP_select_5_earlyvlate, tidy(t.test(.$log_norm_counts ~ .$treatment_timepoint,
                                                              alternative = "two.sided",
                                                              mu = 0,
                                                              var.equal = FALSE,
                                                              conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_5_earlyvlate$p.adj <- p.adjust(t.test_5_earlyvlate$p.value, method = c("fdr"), n = length(VIP_5_earlyvlate$metabolite)) #Adjust p value for the number of comparisons
t.test_5_earlyvlate
```

```{r}
t.test_5_earlyvlate %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  mutate(., higherAbundanceIn = case_when(estimate1 > estimate2 ~ "05C_early",
                                          estimate1 < estimate2 ~ "05C_late")) %>%
  dplyr::select(., metabolite, PubChem, KEGG, higherAbundanceIn) %>%
  unique(.) %>%
  write.table("PLSDA/5-earlyvlate_significantVIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with identifiers. Create a column with information on condition with the higher abundance. Retain unique entries and write table.
```

```{r}
t.test_5_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #42 significantly different metabolites
t.test_5_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #37 metabolites higher earlier
t.test_5_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #5 metabolites higher later
```

```{r}
t.test_5_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_5_earlyvlate, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "Early",
                                  estimate < 0 ~ "Late")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]), 
                    name = "Temperature (ºC)",
                    breaks = c("Early", "Late"),
                    labels = c("Early", "Late")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-5-earlyvlate.pdf", height = 8.5, width = 11)
```

#### Early 25ºC vs. Late 25ºC

```{r}
list <- c("25C_early", "25C_late") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_timepoint) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_25_earlyvlate <- plsda(X, Y, ncomp = 3) #Run PLSDA for desired contrast
VIP_25_earlyvlate <- as.data.frame(PLSDA.VIP(known.plsda_25_earlyvlate)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_25_earlyvlate
```

```{r}
clean_25_earlyvlate <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_timepoint %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_25_earlyvlate <- cbind(clean_25_earlyvlate[, c(1:2, 199:200)], 
                                  log(clean_25_earlyvlate[, names(clean_25_earlyvlate) %in% VIP_25_earlyvlate$metabolite])) %>%
  pivot_longer(., cols = 5:75, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_25_earlyvlate #Confirm changes
```

```{r}
t.test_25_earlyvlate <-do(VIP_select_25_earlyvlate, tidy(t.test(.$log_norm_counts ~ .$treatment_timepoint,
                                                                alternative = "two.sided",
                                                                mu = 0,
                                                                var.equal = FALSE,
                                                                conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_25_earlyvlate$p.adj <- p.adjust(t.test_25_earlyvlate$p.value, method = c("fdr"), n = length(VIP_25_earlyvlate$metabolite)) #Adjust p value for the number of comparisons
t.test_25_earlyvlate
```

```{r}
t.test_25_earlyvlate %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  mutate(., higherAbundanceIn = case_when(estimate1 > estimate2 ~ "25C_early",
                                          estimate1 < estimate2 ~ "25C_late")) %>%
  dplyr::select(., metabolite, PubChem, KEGG, higherAbundanceIn) %>%
  unique(.) %>%
  write.table("PLSDA/25-earlyvlate_significantVIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with identifiers. Create a column with information on condition with the higher abundance. Retain unique entries and write table.
```

```{r}
t.test_25_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #69 significantly different metabolites
t.test_25_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #63 metabolites higher earlier
t.test_25_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #6 metabolites higher later
```

```{r}
t.test_25_earlyvlate %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_25_earlyvlate, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "Early",
                                  estimate < 0 ~ "Late")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c("salmon2", plotColors[1]), 
                    name = "Temperature (ºC)",
                    breaks = c("Early", "Late"),
                    labels = c("Early", "Late")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-25-earlyvlate.pdf", height = 8.5, width = 11)
```

#### Metaboanalyst enrichment

I put the VIP lists from each contrast and put them into the Pathway Analysis module in Metaboanalyst, following specifications listed above.

##### Early 5ºC vs. Early 25ºC

```{r}
early_5v25_pathwayResults <- read.csv("metaboanalyst/5v25-early_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(early_5v25_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
early_5v25_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5v25_early-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_early_5v25_pathwayResults <- early_5v25_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 pathways
head(top_early_5v25_pathwayResults) #Confirm formatting
```

```{r}
top_early_5v25_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_early_5v25_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5v25_early-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Arginine biosynthesis is significantly enriched pathway associated with metabolites that differentiate 5ºC and 25ºC at the earlier timepoint. The compounds in this pathway are:

C00077: ornithine
C00327: citrulline
C00086: urea
C00122: fumaric acid

```{r}
early_5v25_arginineBiosynthesis_facets <- c("1" = "ornithine",
                          "2" = "citrulline",
                          "3" = "urea",
                          "4" = "fumaric acid") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     `ornithine`, `citrulline`, urea, `fumaric acid`)) %>%
  filter(., treatment_timepoint == "05C_early" | treatment_timepoint == "25C_early") %>%
  pivot_longer(., cols = c(3:6), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "ornithine" ~ 1,
                              metabolite == "citrulline" ~ 2,
                              metabolite == "urea" ~ 3,
                              metabolite == "fumaric acid" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = early_5v25_arginineBiosynthesis_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                               "salmon2"),
                    labels = c("Early 5ºC", 
                               "Early 25ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                               "salmon2"),
                    labels = c("Early 5ºC", 
                               "Early 25ºC")) +
  xlab("") + #Axis titles
  ggtitle("Arginine Biosynthesis") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/5v25-early_arginine-biosynthesis-plot.pdf", height = 8.5, width = 11)
```

##### Late 5ºC vs. Late 25ºC

```{r}
late_5v25_pathwayResults <- read.csv("metaboanalyst/5v25-late_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(late_5v25_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
late_5v25_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5v25_late-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_late_5v25_pathwayResults <- late_5v25_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_late_5v25_pathwayResults) #Confirm formatting
```

```{r}
top_late_5v25_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_late_5v25_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5v25_late-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Starch and sucrose metabolism and glutathione metabolism are significantly enriched between these comparisons. I've already plotted starch and sucrose metabolism above, but I'll plot it again here since there are some compounds that are different between those two datasets. I can also remake the graph to focus specifically on the 5ºC vs. 25ºC comparison at the later timepoint.

Starch and sucrose metabolism:

C00208: maltose
C00103: glucose-1-phosphate
C00689: trehalose-6-phosphate
C00031: glucose

```{r}
late_5v25_starchAndSucrose_facets <- c("1" = "maltose",
                          "2" = "glucose-1-phosphate",
                          "3" = "trehalose-6-phosphate",
                          "4" = "glucose") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     maltose, `glucose-1-phosphate`, `trehalose-6-phosphate`, glucose)) %>%
  filter(., treatment_timepoint == "05C_late" | treatment_timepoint == "25C_late") %>%
  pivot_longer(., cols = c(3:6), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "maltose" ~ 1,
                              metabolite == "glucose-1-phosphate" ~ 2,
                              metabolite == "trehalose-6-phosphate" ~ 3,
                              metabolite == "glucose" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = late_5v25_starchAndSucrose_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c(plotColors[3], 
                               plotColors[1]),
                    labels = c("Late 5ºC", 
                               "Late 25ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c(plotColors[3], 
                                plotColors[1]),
                    labels = c("Late 5ºC", 
                               "Late 25ºC")) +
  xlab("") + #Axis titles
  ggtitle("Starch and Sucrose Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/5v25-late_starch-sucrose-metabolism-plot.pdf", height = 8.5, width = 11)
```

Glutathione metabolism:

C00077: ornithine
C01672: cadaverine
C00037: glycine
C00051: glutathione
C01879: oxoproline

```{r}
late_5v25_glutathioneMetabolism_facets <- c("1" = "ornithine",
                          "2" = "cadaverine",
                          "3" = "glycine",
                          "4" = "glutathione",
                          "5" = "oxoproline") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     ornithine, cadaverine, glycine, glutathione, oxoproline)) %>%
  filter(., treatment_timepoint == "05C_late" | treatment_timepoint == "25C_late") %>%
  pivot_longer(., cols = c(3:7), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "ornithine" ~ 1,
                              metabolite == "cadaverine" ~ 2,
                              metabolite == "glycine" ~ 3,
                              metabolite == "glutathione" ~ 4,
                              metabolite == "oxoproline" ~ 5)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = late_5v25_glutathioneMetabolism_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c(plotColors[3], 
                               plotColors[1]),
                    labels = c("Late 5ºC", 
                               "Late 25ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c(plotColors[3], 
                                plotColors[1]),
                    labels = c("Late 5ºC", 
                               "Late 25ºC")) +
  xlab("") + #Axis titles
  ggtitle("Glutathione Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/5v25-late_glutathione-metabolism-plot.pdf", height = 8.5, width = 11)
```

##### Early 5ºC vs. Late 5ºC

```{r}
earlyvlate_5_pathwayResults <- read.csv("metaboanalyst/5-earlyvlate_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(earlyvlate_5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
earlyvlate_5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_earlyvlate-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_earlyvlate_5_pathwayResults <- earlyvlate_5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_earlyvlate_5_pathwayResults) #Confirm formatting
```

```{r}
top_earlyvlate_5_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_earlyvlate_5_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_earlyvlate-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Taurine and hypotaurine metabolism:

C00245: taurine
C00097: cysteine
C00519: hypotaurine

```{r}
earlyvlate_5_taurineMetabolism_facets <- c("1" = "taurine",
                          "2" = "cysteine",
                          "3" = "hypotaurine") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     `taurine`, `cysteine`, `hypotaurine`)) %>%
  filter(., treatment_timepoint == "05C_early" | treatment_timepoint == "05C_late") %>%
  pivot_longer(., cols = c(3:5), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "taurine" ~ 1,
                              metabolite == "cysteine" ~ 2,
                              metabolite == "hypotaurine" ~ 3)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = earlyvlate_5_taurineMetabolism_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                               plotColors[3]),
                    labels = c("Early 5ºC", 
                               "Late 5ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                                plotColors[3]),
                    labels = c("Early 5ºC", 
                               "Late 5ºC")) +
  xlab("") + #Axis titles
  ggtitle("Taurine and Hypotaurine Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/earlyvlate-5_taurine-hypotaurine-metabolism-plot.pdf", height = 8.5, width = 11)
```

Starch and sucrose metabolism:

C00103: glucose-1-phosphate
C00092: glucose-6-phosphate
C00689: trehalose-6-phosphate
C00095: fructose
C00085: fructose-6-phosphate

```{r}
earlyvlate_5_starchAndSucrose_facets <- c("1" = "glucose-1-phosphate",
                          "2" = "glucose-6-phosphate",
                          "3" = "trehalose-6-phosphate",
                          "4" = "fructose",
                          "5" = "fructose-6-phosphate") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     `glucose-1-phosphate`, `glucose-6-phosphate`, `trehalose-6-phosphate`, fructose, `fructose-6-phosphate`)) %>%
  filter(., treatment_timepoint == "05C_early" | treatment_timepoint == "05C_late") %>%
  pivot_longer(., cols = c(3:7), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "glucose-1-phosphate" ~ 1,
                              metabolite == "glucose-6-phosphate" ~ 2,
                              metabolite == "trehalose-6-phosphate" ~ 3,
                              metabolite == "fructose" ~ 4,
                              metabolite == "fructose-6-phosphate" ~ 5)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = earlyvlate_5_starchAndSucrose_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                               plotColors[3]),
                    labels = c("Early 5ºC", 
                               "Late 5ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("lightblue2", 
                                plotColors[3]),
                    labels = c("Early 5ºC", 
                               "Late 5ºC")) +
  xlab("") + #Axis titles
  ggtitle("Starch and Sucrose Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/earlyvlate-5_starch-sucrose-metabolism-plot.pdf", height = 8.5, width = 11)
```

##### Early 25ºC vs. Late 25ºC

```{r}
earlyvlate_25_pathwayResults <- read.csv("metaboanalyst/25-earlyvlate_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(earlyvlate_5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
earlyvlate_25_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/25_earlyvlate-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_earlyvlate_25_pathwayResults <- earlyvlate_25_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 225 metabolites
head(top_earlyvlate_25_pathwayResults) #Confirm formatting
```

```{r}
top_earlyvlate_25_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_earlyvlate_25_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/25_earlyvlate-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Starch and sucrose metabolism:

Starch and sucrose metabolism:

C00103: glucose-1-phosphate
C00092: glucose-6-phosphate
C00689: trehalose-6-phosphate
C00095: fructose
C00208: maltose

```{r}
earlyvlate_25_starchAndSucrose_facets <- c("1" = "glucose-1-phosphate",
                          "2" = "glucose-6-phosphate",
                          "3" = "trehalose-6-phosphate",
                          "4" = "fructose",
                          "5" = "maltose") #Assign labels for facets
```

```{r}
transMetabData_knownTimepoint %>%
  dplyr::select(., c(crab.ID, treatment_timepoint, 
                     `glucose-1-phosphate`, `glucose-6-phosphate`, `trehalose-6-phosphate`, fructose, maltose)) %>%
  filter(., treatment_timepoint == "25C_early" | treatment_timepoint == "25C_late") %>%
  pivot_longer(., cols = c(3:7), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "glucose-1-phosphate" ~ 1,
                              metabolite == "glucose-6-phosphate" ~ 2,
                              metabolite == "trehalose-6-phosphate" ~ 3,
                              metabolite == "fructose" ~ 4,
                              metabolite == "maltose" ~ 5)) %>%
  ggplot(., mapping = aes(x = treatment_timepoint, y = value, color = treatment_timepoint)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = earlyvlate_25_starchAndSucrose_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("salmon2", 
                               plotColors[1]),
                    labels = c("Early 25ºC", 
                               "Late 25ºC")) +
  scale_colour_manual(name = "Temperature (ºC) x Timepoint", 
                    values = c("salmon2", 
                                plotColors[1]),
                    labels = c("Early 25ºC", 
                               "Late 25ºC")) +
  xlab("") + #Axis titles
  ggtitle("Starch and Sucrose Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/earlyvlate-25_starch-sucrose-metabolism-plot.pdf", height = 8.5, width = 11)
```







# Impact of temperature and genotype on metabolomes

The third question I have is understanding if different supergene genotypes employed different metabolomic pathways to withstand chronic temperature stress. I will look at data from day 0 and day 42 since I have balanced genotype information for those timepoints. Additionally, day 0 (15ºC) will highlight if there were any differences in genotype-specific metabolomes prior to heat stress.

## PCA

### All metabolites

```{r}
(transMetabData %>% filter(., day == "0" | day == "42"))[-c(1:13)] #Data from before and after chronic thermal stress
```

```{r}
scaled.pca.TG <- prcomp((transMetabData %>% filter(., day == "0" | day == "42"))[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TG) #Standard deviations for each PC
```

```{r}
pcaPlot7 <- autoplot(scaled.pca.TG, data = (transMetabData %>% filter(., day == "0" | day == "42")), x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "final.genotype", shape = "final.genotype",
                     frame = TRUE, frame.colour = "final.genotype", loadings = FALSE) +
  scale_color_manual(values = rainbow(3),
                     name = "Genotype") +
  scale_fill_manual(values = rainbow(3),
                    name = "Genotype") +
  scale_shape_manual(values = c(1, 2, 0),
                     name = "Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank())
pcaPlot7
ggsave("PCA/figures/all-data-PCA-genotype.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot8 <- autoplot(scaled.pca.TG, data = (transMetabData %>% filter(., day == "0" | day == "42")), x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "presenceC", shape = "presenceC",
                     frame = TRUE, frame.colour = "presenceC", loadings = FALSE) +
  scale_color_manual(values = rainbow(2),
                     name = "Is the C Allele Present?") +
  scale_fill_manual(values = rainbow(2),
                    name = "Is the C Allele Present?") +
  scale_shape_manual(values = c(1, 2, 0),
                     name = "Is the C Allele Present?") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank())
pcaPlot8
ggsave("PCA/figures/all-data-PCA-presenceC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot9 <- autoplot(scaled.pca.TG, data = (transMetabData %>% filter(., day == "0" | day == "42")), x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "presenceT", shape = "presenceT",
                     frame = TRUE, frame.colour = "presenceT", loadings = FALSE) +
  scale_color_manual(values = rainbow(2),
                     name = "Is the T Allele Present?") +
  scale_fill_manual(values = rainbow(2),
                    name = "Is the T Allele Present?") +
  scale_shape_manual(values = c(1, 2, 0),
                     name = "Is the T Allele Present?") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank())
pcaPlot9
ggsave("PCA/figures/all-data-PCA-presenceT.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot10 <- autoplot(scaled.pca.TG, data = (transMetabData %>% filter(., day == "0" | day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                rep(plotColors[2], times = 3),
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               rep(plotColors[2], times = 3),
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 3)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot10
ggsave("PCA/figures/all-data-PCA-temperatureGenotype.pdf", height = 8.5, width = 11)
```

Alright, it seems like genotype doesn't impact metabolome variation at the beginning of the experiment. I'm going to remove the day 0 data, rescale, and look at only the differences in genotype for the 5ºC and 25ºC treatments.

```{r}
(transMetabData %>% filter(., day == "42"))[-c(1:13)] #Data from after chronic thermal stress
```

```{r}
scaled.pca.TGNC <- prcomp((transMetabData %>% filter(., day == "42"))[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TGNC) #Standard deviations for each PC
```

```{r}
pcaPlot11 <- autoplot(scaled.pca.TGNC, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 2)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot11
ggsave("PCA/figures/all-data-PCA-temperatureGenotypeNC.pdf", height = 8.5, width = 11)
```

While there is a good amount of overlap between the genotypes, it looks like there may be dispersion differences associated with the alleles. In 5ºC, CC and CT crabs had more dispersion whereas TT crabs did not, and CT and TT crabs had more dispersion in 25ºC than the CC crabs. This is interesting considering CC are homozygous for the cold-adapted supergene and TT are homozygous for the warm-adapted supergene.

```{r}
pcaPlot12 <- autoplot(scaled.pca.TGNC, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC",
                                "TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("TT at 5ºC", "CC or CT at 5ºC",
                               "TT at 25ºC", "CC or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC",
                                "TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot12
ggsave("PCA/figures/all-data-PCA-temperaturepresenceCNC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot12 <- autoplot(scaled.pca.TGNC, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC",
                               "CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot12
ggsave("PCA/figures/all-data-PCA-temperaturepresenceTNC.pdf", height = 8.5, width = 11)
```

### Identified metabolites

```{r}
(transMetabData %>% filter(., day == "0" | day == "42"))[c(14:198)] #Data from before and after chronic thermal stress
```

```{r}
scaled.pca.TG.id <- prcomp((transMetabData %>% filter(., day == "0" | day == "42"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TG.id) #Standard deviations for each PC
```

```{r}
pcaPlot13 <- autoplot(scaled.pca.TG.id, data = (transMetabData %>% filter(., day == "0" | day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                rep(plotColors[2], times = 3),
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               rep(plotColors[2], times = 3),
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 3)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 15ºC", "CT at 15ºC", "TT at 15ºC",
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot13
ggsave("PCA/figures/known-metabolites-PCA-temperatureGenotype.pdf", height = 8.5, width = 11)
```

Similar to the plot with all the data, the difference between 15ºC and the other two treatments is dominating the PCA.

```{r}
(transMetabData %>% filter(., day == "42"))[c(14:198)] #Data from after chronic thermal stress
```

```{r}
scaled.pca.TGNC.id <- prcomp((transMetabData %>% filter(., day == "42"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.TGNC.id) #Standard deviations for each PC
```

```{r}
pcaPlot14 <- autoplot(scaled.pca.TGNC.id, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 2)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot14
ggsave("PCA/figures/known-metabolites-PCA-temperatureGenotypeNC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot15 <- autoplot(scaled.pca.TGNC.id, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC",
                                "TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("TT at 5ºC", "CC or CT at 5ºC",
                               "TT at 25ºC", "CC or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC",
                                "TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot15
ggsave("PCA/figures/known-metabolites-PCA-temperaturepresenceCNC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot16 <- autoplot(scaled.pca.TGNC.id, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC",
                               "CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot16
ggsave("PCA/figures/known-metabolites-PCA-temperaturepresenceTNC.pdf", height = 8.5, width = 11)
```

##### Check outlier

Looking at these plots, there appears to be an outlier CC crab at 5ºC.

```{r}
autoplot(scaled.pca.TGNC.id, data = (transMetabData %>% filter(., day == "42")), x = 1, y = 2,
         size = 0, alpha = 0.8, colour = "treatment_genotype", 
         frame = TRUE, frame.colour = "treatment_genotype", 
         loadings = FALSE, label = TRUE, label.label = "crab.ID") +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank())
```

This is crab 5-075. I'm going to rerun the PCA without this crab to see if that changes any of the trends.

```{r}
scaled.pca.TGNC.id.outlier <- prcomp((transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation without the potential outlier
summary(scaled.pca.TGNC.id.outlier) #Standard deviations for each PC
```

```{r}
pcaPlot17 <- autoplot(scaled.pca.TGNC.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                                "salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                               "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 2)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC", 
                                "CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot17
ggsave("PCA/figures/known-metabolites-PCA-temperatureGenotypeNC-no5075.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot18 <- autoplot(scaled.pca.TGNC.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC",
                               "CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot18
ggsave("PCA/figures/known-metabolites-PCA-temperaturePresenceCNC-no5075.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot17 <- autoplot(scaled.pca.TGNC.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC",
                               "CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC",
                                "CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot17
ggsave("PCA/figures/known-metabolites-PCA-temperaturePresenceTNC-no5075.pdf", height = 8.5, width = 11)
```

Without the outlier present, I'm also seeming some differences in dispersion between CT and homozygotes. I think this would be interesting to plot.

```{r}
pcaPlot18 <- autoplot(scaled.pca.TGNC.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_het", shape = "treatment_het", 
                      frame = TRUE, frame.colour = "treatment_het", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3],
                                "salmon2", plotColors[1]),
                     labels = c("CC or TT at 5ºC", "CT at 5ºC",
                                "CC or TT at 25ºC", "CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3],
                               "salmon2", plotColors[1]),
                    labels = c("CC or TT at 5ºC", "CT at 5ºC",
                               "CC or TT at 25ºC", "CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 2)),
                     labels = c("CC or TT at 5ºC", "CT at 5ºC",
                                "CC or TT at 25ºC", "CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot18
ggsave("PCA/figures/known-metabolites-PCA-temperatureHetNC-no5075.pdf", height = 8.5, width = 11)
```

## Conduct PERMANOVA

Given that the dispersion differences are clearer when I do not include 15ºC data, I'm going to only run the tests with 5ºC and 25ºC only. I am also going to look at the impact of genotype, precense of the C allele, and presence of the T allele.

### All metabolites

```{r}
(transMetabData %>% filter(., day == "42"))[-c(1:13)] #Data from after chronic thermal stress
```

```{r}
dissim.TG <- vegdist(scale((transMetabData %>% filter(., day == "42"))[-c(1:13)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

#### Full genotype

```{r}
permanova.TGFull <- adonis2(dissim.TG ~ as.factor(treatment)*as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGFull #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGFull.temp <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.temp) #Dispersion differences exist
boxplot(disp.TGFull.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGFull.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

Even though I didn't see significant centroid differences with genotype, I want to test the dispersion differences since I saw some differences there in the PCA.

```{r}
disp.TGFull.gen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.gen)
boxplot(disp.TGFull.gen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGFull.gen) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGFull.tempGen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.tempGen)
boxplot(disp.TGFull.tempGen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGFull.tempGen) #Variance is not different between groups. No differences in dispersion.
```

#### Presence C

```{r}
permanova.TGpresenceC <- adonis2(dissim.TG ~ as.factor(treatment)*as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceC #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGpresenceC.temp <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.temp) #Dispersion differences exist
boxplot(disp.TGpresenceC.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceC.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGpresenceC.gen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.gen)
boxplot(disp.TGpresenceC.gen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceC.gen) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGpresenceC.tempGen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.tempGen)
boxplot(disp.TGpresenceC.tempGen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceC.tempGen) #Variance is not different between groups. No differences in dispersion.
```

#### Presence T

```{r}
permanova.TGpresenceT <- adonis2(dissim.TG ~ as.factor(treatment)*as.factor(presenceT), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceT #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGpresenceT.temp <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.temp) #Dispersion differences exist
boxplot(disp.TGpresenceT.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceT.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGpresenceT.gen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.gen)
boxplot(disp.TGpresenceT.gen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceT.gen) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGpresenceT.tempGen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.tempGen)
boxplot(disp.TGpresenceT.tempGen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGpresenceT.tempGen) #Variance is not different between groups. No differences in dispersion.
```

#### Heterozygote vs. homozygote

```{r}
permanova.TGHet <- adonis2(dissim.TG ~ as.factor(treatment)*as.factor(het), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGHet #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGHet.temp <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGHet.temp) #Dispersion differences exist
boxplot(disp.TGHet.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGHet.temp) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGHet.gen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGHet.gen)
boxplot(disp.TGHet.gen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGHet.gen) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGHet.tempGen <- betadisper(dissim.TG, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGHet.tempGen)
boxplot(disp.TGHet.tempGen) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGHet.tempGen) #Variance is not different between groups. No differences in dispersion.
```

#### Save test output

```{r}
rbind(
  broom::tidy(permanova.TGFull) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.TGpresenceC) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.TGpresenceT) %>%
    mutate(genotype = "presenceT"), 
  broom::tidy(permanova.TGHet) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv(., "PCA/all-data-PERMANOVA-temperatureGenotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.TGFull.temp)) %>%
    mutate(genotype = "full",
           var = "temperature"),
  broom::tidy(anova(disp.TGFull.gen)) %>%
    mutate(genotype = "full",
           var = "genotype"),
  broom::tidy(anova(disp.TGFull.tempGen)) %>%
    mutate(genotype = "full",
           var = "interaction"),
  broom::tidy(anova(disp.TGpresenceC.temp)) %>%
    mutate(genotype = "presenceC",
           var = "temperature"),
  broom::tidy(anova(disp.TGpresenceC.gen)) %>%
    mutate(genotype = "presenceC",
           var = "genotype"),
  broom::tidy(anova(disp.TGpresenceC.tempGen)) %>%
    mutate(genotype = "presenceC",
           var = "interaction"),
  broom::tidy(anova(disp.TGpresenceT.temp)) %>%
    mutate(genotype = "presenceT",
           var = "temperature"),
  broom::tidy(anova(disp.TGpresenceT.gen)) %>%
    mutate(genotype = "presenceT",
           var = "genotype"),
  broom::tidy(anova(disp.TGpresenceT.tempGen)) %>%
    mutate(genotype = "presenceT",
           var = "interaction"),
  broom::tidy(anova(disp.TGHet.temp)) %>%
    mutate(genotype = "het",
           var = "temperature"),
  broom::tidy(anova(disp.TGHet.gen)) %>%
    mutate(genotype = "het",
           var = "genotype"),
  broom::tidy(anova(disp.TGHet.tempGen)) %>%
    mutate(genotype = "het",
           var = "interaction") 
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/all-data-PERMDISP-temperatureGenotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

### Identified metabolites

```{r}
dissim.TG.id <- vegdist(scale((transMetabData %>% filter(., day == "42"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

#### Full genotype

```{r}
permanova.TGFull.id <- adonis2(dissim.TG.id ~ as.factor(treatment)*as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGFull.id #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGFull.temp.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.temp.id) #PCoA
boxplot(disp.TGFull.temp.id) #Show distance to group centroids
anova(disp.TGFull.temp.id) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGFull.gen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.gen.id)
boxplot(disp.TGFull.gen.id) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGFull.gen.id) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGFull.tempGen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGFull.tempGen.id)
boxplot(disp.TGFull.tempGen.id) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.TGFull.tempGen.id) #Variance is not different between groups. No differences in dispersion.
```

#### Presence C

```{r}
permanova.TGpresenceC.id <- adonis2(dissim.TG.id ~ as.factor(treatment)*as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceC.id #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGpresenceC.temp.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.temp.id) #PCoA
boxplot(disp.TGpresenceC.temp.id) #Show distance to group centroids
anova(disp.TGpresenceC.temp.id) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGpresenceC.gen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.gen.id) #PCoA
boxplot(disp.TGpresenceC.gen.id) #Show distance to group centroids
anova(disp.TGpresenceC.gen.id) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGpresenceC.tempGen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceC.tempGen.id) #PCoA
boxplot(disp.TGpresenceC.tempGen.id) #Show distance to group centroids
anova(disp.TGpresenceC.tempGen.id) #Variance is not different between groups. No differences in dispersion.
```

#### Presence T

```{r}
permanova.TGpresenceT.id <- adonis2(dissim.TG.id ~ as.factor(treatment)*as.factor(presenceT), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceT.id #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGpresenceT.temp.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.temp.id) #PCoA
boxplot(disp.TGpresenceT.temp.id) #Show distance to group centroids
anova(disp.TGpresenceT.temp.id) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGpresenceT.gen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.gen.id) #PCoA
boxplot(disp.TGpresenceT.gen.id) #Show distance to group centroids
anova(disp.TGpresenceT.gen.id) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGpresenceT.tempGen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGpresenceT.tempGen.id) #PCoA
boxplot(disp.TGpresenceT.tempGen.id) #Show distance to group centroids
anova(disp.TGpresenceT.tempGen.id) #Variance is not different between groups. No differences in dispersion.
```

#### Heterozygote vs. homozygote

```{r}
permanova.TGhet.id <- adonis2(dissim.TG.id ~ as.factor(treatment)*as.factor(het), data = (transMetabData %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGhet.id #Significant influence of temperature, but not genotype or the interaction
```

```{r}
disp.TGhet.temp.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGhet.temp.id) #PCoA
boxplot(disp.TGhet.temp.id) #Show distance to group centroids
anova(disp.TGhet.temp.id) #Variance is not different between groups. Temperature significance is due to centroid differences only
```

```{r}
disp.TGhet.gen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGhet.gen.id) #PCoA
boxplot(disp.TGhet.gen.id) #Show distance to group centroids
anova(disp.TGhet.gen.id) #Variance is not different between groups. No differences in dispersion.
```

```{r}
disp.TGhet.tempGen.id <- betadisper(dissim.TG.id, group = as.factor((transMetabData %>% filter(., day == "42"))$treatment_het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.TGhet.tempGen.id) #PCoA
boxplot(disp.TGhet.tempGen.id) #Show distance to group centroids
anova(disp.TGhet.tempGen.id) #Variance is not different between groups. No differences in dispersion.
```

#### Check outlier removal impact on test results

I identified a potential outlier in sample 5-075. I will see if outlier removal impacts the significance of any of the variables tested above. If it does, then I will move forward with that statistical output and outlier removal. If it does not, then everything I've done so far is fine!

```{r}
dissim.TG.id.outlier <- vegdist(scale((transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.TGFull.id.outlier <- adonis2(dissim.TG.id.outlier ~ as.factor(treatment)*as.factor(final.genotype), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGFull.id.outlier #Significant influence of temperature, but not genotype or the interaction
```

```{r}
permanova.TGpresenceC.id.outlier <- adonis2(dissim.TG.id.outlier ~ as.factor(treatment)*as.factor(presenceC), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceC.id.outlier #Significant influence of temperature, but not genotype or the interaction
```

```{r}
permanova.TGpresenceT.id.outlier <- adonis2(dissim.TG.id.outlier ~ as.factor(treatment)*as.factor(presenceT), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGpresenceT.id.outlier #Significant influence of temperature, but not genotype or the interaction
```

```{r}
permanova.TGHet.id.outlier <- adonis2(dissim.TG.id.outlier ~ as.factor(treatment)*as.factor(het), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.TGHet.id.outlier #Significant influence of temperature, but not genotype or the interaction
```

There is no significant impact of outlier removal on these tests, so I'll move forward with the statistical results that include the outlier.

#### Save test output

```{r}
rbind(
  broom::tidy(permanova.TGFull.id) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.TGpresenceC.id) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.TGpresenceT.id) %>%
    mutate(genotype = "presenceT"),
  broom::tidy(permanova.TGhet.id) %>%
    mutate(genotype = "het")
) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-temperatureGenotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.TGFull.temp.id)) %>%
    mutate(genotype = "full",
           var = "temperature"),
  broom::tidy(anova(disp.TGFull.gen.id)) %>%
    mutate(genotype = "full",
           var = "genotype"),
  broom::tidy(anova(disp.TGFull.tempGen.id)) %>%
    mutate(genotype = "full",
           var = "interaction"),
  broom::tidy(anova(disp.TGpresenceC.temp.id)) %>%
    mutate(genotype = "presenceC",
           var = "temperature"),
  broom::tidy(anova(disp.TGpresenceC.gen.id)) %>%
    mutate(genotype = "presenceC",
           var = "genotype"),
  broom::tidy(anova(disp.TGpresenceC.tempGen.id)) %>%
    mutate(genotype = "presenceC",
           var = "interaction"),
  broom::tidy(anova(disp.TGpresenceT.temp.id)) %>%
    mutate(genotype = "presenceT",
           var = "temperature"),
  broom::tidy(anova(disp.TGpresenceT.gen.id)) %>%
    mutate(genotype = "presenceT",
           var = "genotype"),
  broom::tidy(anova(disp.TGpresenceT.tempGen.id)) %>%
    mutate(genotype = "presenceT",
           var = "interaction"),
  broom::tidy(anova(disp.TGhet.temp.id)) %>%
    mutate(genotype = "het",
           var = "temperature"),
  broom::tidy(anova(disp.TGhet.gen.id)) %>%
    mutate(genotype = "het",
           var = "genotype"),
  broom::tidy(anova(disp.TGhet.tempGen.id)) %>%
    mutate(genotype = "het",
           var = "interaction")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-temperatureGenotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

Moving forward, I will consider interactions between temperature and full genotype since that gives me the most balanced sample sizes.

## ASCA: Temperature x full genotype

### Format data

```{r}
tempGen.ASCA <- transMetabData[c(1:198)] %>%
  filter(., day == "42") %>%
  dplyr::select(-c(treatment:treatment_het)) %>%
  pivot_longer(., cols = !crab.ID, names_to = "variable", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  left_join(., metabolomicsMetadata, by = "crab.ID") %>%
  left_join(., genotypeData, by = "crab.ID") %>%
  dplyr::rename(treatment = treatment.x,
                original.treatment = treatment.y) %>%
  dplyr::select(., c(crab.ID:treatment, final.genotype)) %>%
  mutate(., treatment = as.factor(treatment),
         final.genotype = as.factor(final.genotype))
#Take known metabolomics data and filter day 0 information. Rmeove extraneous columns. Pivot data longer for ASCA. Compound information should be "variable." Add metadata to dataframe with all compound information and retain only useful columns (treatment, day, and final genotype). Convert variables to factors.
head(tempGen.ASCA) #Confirm changes
```

### Run model

```{r}
tempGen.ASCAmodel <- ALASCA(df = tempGen.ASCA,
                            formula = value ~ final.genotype*treatment + (1|crab.ID),
                            n_validation_runs = 100,
                            validate = TRUE,
                            reduce_dimensions = TRUE) #Run ASCA to identify day and temperature effects on compound abundance. Include individual crab as a random effect
```

### Interpret ASCA

```{r}
# pdf("ASCA/figures/tempGen-scree-plot.pdf")
plot(tempGen.ASCAmodel, component = 1:6, type = "scree") #Scree plot of variance explained by each PC. There's a pretty steep drop-off between components 1 and 2.
# dev.off()
```

#### Component 1

This component shows metabolites with consistently higher abundance at 25ºC with slightly higher abundance in TT crabs.

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 1, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-effect-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 1, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-prediction-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
tempGen.loadingsPC1 <- get_loadings(tempGen.ASCAmodel, component = 1, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempGen.loadingsPC1)
```

```{r}
tempGen.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempGen-loadings-PC1.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 2

This component shows metabolites with higher abundance in CC crabs at 5ºC and lower abundance in CC crabs at 25ºC.

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 2, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-effect-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 2, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-prediction-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
tempGen.loadingsPC2 <- get_loadings(tempGen.ASCAmodel, component = 2, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempGen.loadingsPC2)
```

```{r}
tempGen.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempGen-loadings-PC2.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

#### Component 3

This component shows metabolites with higher abundance in TT crabs at 5ºC and higher abundance in CT crabs at 25ºC.

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 3, type = 'effect', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-effect-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
plot(tempGen.ASCAmodel, effect = 1, component = 3, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = c(plotColors[3], plotColors[1]),
     my_theme = theme_classic(),
     x_label = "Genotype",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/tempGen-prediction-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
tempGen.loadingsPC3 <- get_loadings(tempGen.ASCAmodel, component = 3, effect = 1) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(tempGen.loadingsPC3)
```

```{r}
tempGen.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = "covars") %>%
  write.csv("ASCA/tempGen-loadings-PC3.csv", quote = FALSE, row.names = FALSE) #Save loading information
```

### Metaboanalyst

#### Component 1

```{r}
tempGen.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempGen.loadingsPC1[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("PLSDA/tempGen-PC1_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempGen_PC1_pathwayResults <- read.csv("metaboanalyst/tempGen-loadings-PC1_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempGen_PC1_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempGen_PC1_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempGen_PC1-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_tempGen_PC1_pathwayResults <- tempGen_PC1_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 pathways
head(top_tempGen_PC1_pathwayResults) #Confirm formatting
```

```{r}
top_tempGen_PC1_pathwayResults %>%
  mutate(., order = rev(1:14)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_tempGen_PC1_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempGen_PC1-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Glutathione metabolism:

C01672: cadaverine
C00037: glycine
C00051: glutathione
C01879: oxoproline

```{r}
tempGen_PC1_glutathioneMetabolism_facets <- c("1" = "cadaverine",
                          "2" = "glycine",
                          "3" = "glutathione",
                          "4" = "oxoproline") #Assign labels for facets
```

```{r}
transMetabData %>%
  filter(., treatment != "15C") %>%
  dplyr::select(., c(crab.ID, treatment_genotype, 
                     cadaverine, glycine, glutathione, oxoproline)) %>%
  pivot_longer(., cols = c(3:6), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "cadaverine" ~ 1,
                              metabolite == "glycine" ~ 2,
                              metabolite == "glutathione" ~ 3,
                              metabolite == "oxoproline" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_genotype, y = value, color = treatment_genotype)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = tempGen_PC1_glutathioneMetabolism_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Genotype", 
                    values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC CC", "5ºC CT", "5ºC TT", 
                               "25ºC CC", "25ºC CT", "25ºC TT")) +
  scale_colour_manual(name = "Temperature (ºC) x Genotype", 
                    values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC CC", "5ºC CT", "5ºC TT", 
                               "25ºC CC", "25ºC CT", "25ºC TT")) +
  xlab("") + #Axis titles
  ggtitle("Glutathione Metabolism") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/tempGen-PC1_glutathione-metabolism-plot.pdf", height = 8.5, width = 11)
```

#### Component 2

```{r}
tempGen.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempGen.loadingsPC2[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("PLSDA/tempGen-PC2_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempGen_PC2_pathwayResults <- read.csv("metaboanalyst/tempGen-loadings-PC2_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempGen_PC2_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempGen_PC2_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempGen_PC2-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_tempGen_PC2_pathwayResults <- tempGen_PC2_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 pathways
head(top_tempGen_PC2_pathwayResults) #Confirm formatting
```

```{r}
top_tempGen_PC2_pathwayResults %>%
  mutate(., order = rev(1:24)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_tempGen_PC2_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempGen_PC2-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

Arginine biosynthesis:

C00077: ornithine
C00327: citrulline
C00026: alpha-ketoglutarate
C00064: glutamine

```{r}
tempGen_PC2_arginineBiosynthesis_facets <- c("1" = "ornithine",
                          "2" = "citrulline",
                          "3" = "alpha-ketoglutarate",
                          "4" = "glutamine") #Assign labels for facets
```

```{r}
transMetabData %>%
  filter(., treatment != "15C") %>%
  dplyr::select(., c(crab.ID, treatment_genotype, 
                     ornithine, citrulline, `alpha-ketoglutarate`, glutamine)) %>%
  pivot_longer(., cols = c(3:6), names_to = "metabolite", values_to = "value") %>%
  mutate(value = log(value)) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) %>%
  mutate(., order = case_when(metabolite == "ornithine" ~ 1,
                              metabolite == "citrulline" ~ 2,
                              metabolite == "alpha-ketoglutarate" ~ 3,
                              metabolite == "glutamine" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_genotype, y = value, color = treatment_genotype)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = tempGen_PC2_arginineBiosynthesis_facets)) +
  ylab("Log-Transformed Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Genotype", 
                    values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC CC", "5ºC CT", "5ºC TT", 
                               "25ºC CC", "25ºC CT", "25ºC TT")) +
  scale_colour_manual(name = "Temperature (ºC) x Genotype", 
                    values = c("lightblue2", "steelblue1", plotColors[3],
                               "salmon2", "firebrick1", plotColors[1]),
                    labels = c("5ºC CC", "5ºC CT", "5ºC TT", 
                               "25ºC CC", "25ºC CT", "25ºC TT")) +
  xlab("") + #Axis titles
  ggtitle("Arginine Biosynthesis") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/tempGen-PC2_arginine-biosynthesis-plot.pdf", height = 8.5, width = 11)
```

#### Component 3

```{r}
tempGen.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  ggplot(., aes(x = rank, y = cum_var)) +
  geom_point(size = 0.7) +
  geom_hline(yintercept = 0.50, linetype = "dashed") + 
  labs(x = "Rank (|Loading|)", y = "Cumulative Proportion of Variance") +
  theme_classic() + theme(base_size = 15) #Take loading information and get statistics of interest (absolute value of loadings, rank, variance contributed, cumulative variance contributed). Plot graph showing where 50% cutoff is
```

```{r}
tempGen.loadingsPC3[[1]] %>%
  dplyr::rename(metabolite = covars) %>%
  mutate(abs_loading = abs(loading)) %>%
  arrange(desc(abs_loading)) %>%
  mutate(rank = 1:nrow(.),
         cum_abs = cumsum(abs_loading),
         var_contrib = loading^2,
         cum_var = cumsum(var_contrib) / sum(var_contrib)) %>%
  filter(., cum_var <= 0.50) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., PC, metabolite, loading, cum_var, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("PLSDA/tempGen-PC3_cumvar50_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Save list of metabolites to use with Metaboanalyst
```

```{r}
tempGen_PC3_pathwayResults <- read.csv("metaboanalyst/tempGen-loadings-PC3_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(tempGen_PC3_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
tempGen_PC3_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/tempGen_PC3-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

No significant pathways, will not be proceeding with further visualization.

## PLS-DA: Temperature x full gneotype

### Format data

```{r}
X_tempGen <- transMetabData %>% 
  filter(., day == "42") %>%
  dplyr::select(c(14:198)) %>%
  log(.) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) #Assign known metabolite data to the X dataframe after log transformation. Replace any -Inf with NA, then replace NA with 0
Y_tempGen <- as.factor((transMetabData %>% filter(., day == "42"))$treatment_genotype) #Assign treatment data to the Y dataframe
```

```{r}
nzv_indices <- nearZeroVar(X_tempGen) #Check to see if any variables have near-zero variance.
nzv_indices$Position #There aren't any columns with near-zero variance!
```

### Tune PLS-DA model

```{r}
tempGen.known.plsda <- plsda(X_tempGen, Y_tempGen, ncomp = 10) #Initially set a higher number for the number of components since we will be testing this
plotIndiv(tempGen.known.plsda) #Preliminary PLS-DA
```

```{r}
perf.tempGen.known.plsda <- perf(tempGen.known.plsda, validation = "Mfold", folds = 4,  
                                 progressBar = TRUE, nrepeat = 50) 

plot(perf.tempGen.known.plsda, sd = TRUE) #Overall, increasing the number of components decreases the error rate. Lowest error rates are for 4 components
perf.tempGen.known.plsda$choice.ncomp #Confirm information from plot.
```

### Run PLS-DA

```{r}
tempGen.known.plsda.tuned <- plsda(X_tempGen, Y_tempGen, ncomp = 4) #Run PLS-DA with the appropriate number of components to include based on tuning above.
plotIndiv(tempGen.known.plsda.tuned, comp = c(1,2)) #Preliminary PLS-DA.
plotIndiv(tempGen.known.plsda.tuned, comp = c(1,3)) #Preliminary PLS-DA.
plotIndiv(tempGen.known.plsda.tuned, comp = c(1,4)) #Preliminary PLS-DA.
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-tempGenotype-comp12.pdf", height = 8.5, width = 11)
plotIndiv(tempGen.known.plsda.tuned, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", "steelblue1", plotColors[3],
                  "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(c(rep(1, 2, 0), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE,
          title = "PLS-DA by Temperature and Genotype", X.label = "Component 1: 15% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 7% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Do not include a legend. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

### Validate PLS-DA

```{r}
tune.splsda.tempGen.known <- tune.splsda(X_tempGen, Y_tempGen, ncomp = 4, 
                                         validation = 'Mfold', folds = 4,
                                         measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.tempGen.known) #Plot results
```

```{r}
error <- tune.splsda.tempGen.known$error.rate #Save error rate
ncomp <- tune.splsda.tempGen.known$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #2 component is the optimal number according to this analysis. However, looking at the error rate plot, error rate is much lower when using 4 components.

select.keepX <- tune.splsda.tempGen.known$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with this many features per component.
```


```{r}
final.splsda.tempGen.known <- splsda(X_tempGen, Y_tempGen, ncomp = 4, keepX = select.keepX) #Use the number of components and metabolites to keep based on optimization from above
plotIndiv(final.splsda.tempGen.known, comp = c(1,2))
plotIndiv(final.splsda.tempGen.known, comp = c(1,3))
plotIndiv(final.splsda.tempGen.known, comp = c(1,4))

#The % variance explained by PCs 3 and 4 is higher than PC 2. This could be because the sPLS identified 2 as the lowest error rate, but I still made my sPLSDA with 4 components. I'll only consider the first two components.
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-tempGen-comp12.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.tempGen.known, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", "steelblue1", plotColors[3],
                  "salmon2", "firebrick1", plotColors[1]),
          pch = as.factor(rep(c(1, 2, 0), times = 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE,
          title = "sPLS-DA by Temperature and Genotype", X.label = "Component 1: 12% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 9% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

### Extract VIP

```{r}
treatmentGenotype.VIP <- as.data.frame(PLSDA.VIP(tempGen.known.plsda.tuned)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") #Extract VIP information and save as a dataframe. Convert metabolite rownames to a column.
head(treatmentGenotype.VIP) #Confirm changes
```

```{r}
write.csv(treatmentGenotype.VIP, "PLSDA/known-metabolites-VIP-treatmentGenotype.csv", quote = FALSE, row.names = FALSE) #Save VIP scores
```

```{r}
treatmentGenotype.VIP %>%
  filter(VIP >= 1) %>%
  nrow(.) #76 metabolites have VIP > 1
```

```{r}
treatmentGenotype.VIP %>%
  filter(VIP >= 1) %>%
  arrange(VIP) %>%
  ggplot(aes(x = VIP, y = reorder(metabolite, VIP, sum))) +
  geom_point() +
  xlab("VIP Score") + ylab("Metabolite") +
  theme_classic() + theme(axis.text.y = element_text(size = 8)) #Using only metabolites with VIP > 1, arrange by metabolite name and reorder by cumulative VIP score. Plot points and change x and y axis labels.
ggsave("PLSDA/figures/known-metabolites-VIP-treatmentGenotype.pdf", height = 8.5, width = 11)
```

### Pairwise tests for VIP

```{r}
transMetabData_knownTimepoint$treatment_genotype %>% as.factor() %>% levels() #List of descriptors for different treatments
```

Based on the PLS-DA and ASCA, I think the informative comparisons are as follows:

- 05C_CC vs. 05C_CT (5ºC CC crabs separate from CT and TT in the PLSDA and sPLSDA, CC exhibits different trends in ASCA PC 2)
- 05C_CC vs. 05C_TT (5ºC CC crabs separate from CT and TT in the PLSDA and sPLSDA, CC exhibits different trends in ASCA PC 2)
- 025C_CC vs. 025C_CT (25ºC CC crabs separate from CT and TT in the sPLSDA, CC exhibits different trends in ASCA PC 2, CT exhibits different trends in ASCA PC 3)
- 025C_CC VS. 025C_TT (25ºC CC crabs separate from CT and TT in the sPLSDA, CC exhibits different trends in ASCA PC 2)
- 025C_CT vs. 025C_TT (25ºC CT exhibits different trends in ASCA PC 3)

#### 5ºC: CT vs. TT

This is not an informative trend according to my PLSDA and ASCA, but I want to confirm that there are minimal differences between CT and TT crabs at 5ºC. If there are any VIP identified, I'll remove these compounds from the informative comparisons so I'm only focusing on differences unique to the comparisons of interest.

```{r}
list <- c("05C_CT", "05C_TT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CTvTT_05C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CTvTT_05C <- as.data.frame(PLSDA.VIP(known.plsda_CTvTT_05C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CTvTT_05C
```

```{r}
clean_CTvTT_05C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CTvTT_05C <- cbind(clean_CTvTT_05C[, c(1:2, 6, 10)], 
                              log(clean_CTvTT_05C[, names(clean_CTvTT_05C) %in% VIP_CTvTT_05C$metabolite])) %>%
  pivot_longer(., cols = 5:76, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CTvTT_05C #Confirm changes
```

```{r}
t.test_CTvTT_05C <-do(VIP_select_CTvTT_05C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CTvTT_05C$p.adj <- p.adjust(t.test_CTvTT_05C$p.value, method = c("fdr"), n = length(VIP_CTvTT_05C$metabolite)) #Adjust p value for the number of comparisons
t.test_CTvTT_05C
```

```{r}
t.test_CTvTT_05C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

Confirmed that there are no differences between these two conditions, which is what I would expect based on the ASCA and PLSDA!

#### 5ºC: CC vs. CT

```{r}
list <- c("05C_CC", "05C_CT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CCvCT_05C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CCvCT_05C <- as.data.frame(PLSDA.VIP(known.plsda_CCvCT_05C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CCvCT_05C
```

```{r}
clean_CCvCT_05C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CCvCT_05C <- cbind(clean_CCvCT_05C[, c(1:2, 6, 10)], 
                              log(clean_CCvCT_05C[, names(clean_CCvCT_05C) %in% VIP_CCvCT_05C$metabolite])) %>%
  pivot_longer(., cols = 5:72, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CCvCT_05C #Confirm changes
```

```{r}
t.test_CCvCT_05C <-do(VIP_select_CCvCT_05C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CCvCT_05C$p.adj <- p.adjust(t.test_CCvCT_05C$p.value, method = c("fdr"), n = length(VIP_CCvCT_05C$metabolite)) #Adjust p value for the number of comparisons
t.test_CCvCT_05C
```

```{r}
t.test_CCvCT_05C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

No significant VIP driving differences between CC and CT crabs.

#### 5ºC: CC vs. TT

```{r}
list <- c("05C_CC", "05C_TT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CCvTT_05C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CCvTT_05C <- as.data.frame(PLSDA.VIP(known.plsda_CCvTT_05C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CCvTT_05C
```

```{r}
clean_CCvTT_05C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CCvTT_05C <- cbind(clean_CCvTT_05C[, c(1:2, 6, 10)], 
                              log(clean_CCvTT_05C[, names(clean_CCvTT_05C) %in% VIP_CCvTT_05C$metabolite])) %>%
  pivot_longer(., cols = 5:76, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CCvTT_05C #Confirm changes
```

```{r}
t.test_CCvTT_05C <-do(VIP_select_CCvTT_05C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CCvTT_05C$p.adj <- p.adjust(t.test_CCvTT_05C$p.value, method = c("fdr"), n = length(VIP_CCvTT_05C$metabolite)) #Adjust p value for the number of comparisons
t.test_CCvTT_05C
```

```{r}
t.test_CCvTT_05C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 25ºC: CC vs. CT

```{r}
list <- c("25C_CC", "25C_CT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CCvCT_25C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CCvCT_25C <- as.data.frame(PLSDA.VIP(known.plsda_CCvCT_25C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CCvCT_25C
```

```{r}
clean_CCvCT_25C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CCvCT_25C <- cbind(clean_CCvCT_25C[, c(1:2, 6, 10)], 
                              log(clean_CCvCT_25C[, names(clean_CCvCT_25C) %in% VIP_CCvCT_25C$metabolite])) %>%
  pivot_longer(., cols = 5:76, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CCvCT_25C #Confirm changes
```

```{r}
t.test_CCvCT_25C <-do(VIP_select_CCvCT_25C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CCvCT_25C$p.adj <- p.adjust(t.test_CCvCT_25C$p.value, method = c("fdr"), n = length(VIP_CCvCT_25C$metabolite)) #Adjust p value for the number of comparisons
t.test_CCvCT_25C
```

```{r}
t.test_CCvCT_25C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 25ºC: CC vs. TT

```{r}
list <- c("25C_CC", "25C_CT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CCvTT_25C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CCvTT_25C <- as.data.frame(PLSDA.VIP(known.plsda_CCvTT_25C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CCvTT_25C
```

```{r}
clean_CCvTT_25C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CCvTT_25C <- cbind(clean_CCvTT_25C[, c(1:2, 6, 10)], 
                              log(clean_CCvTT_25C[, names(clean_CCvTT_25C) %in% VIP_CCvTT_25C$metabolite])) %>%
  pivot_longer(., cols = 5:77, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CCvTT_25C #Confirm changes
```

```{r}
t.test_CCvTT_25C <-do(VIP_select_CCvTT_25C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CCvTT_25C$p.adj <- p.adjust(t.test_CCvTT_25C$p.value, method = c("fdr"), n = length(VIP_CCvTT_25C$metabolite)) #Adjust p value for the number of comparisons
t.test_CCvTT_25C
```

```{r}
t.test_CCvTT_25C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 25ºC: CT vs. TT

```{r}
list <- c("25C_CT", "25C_TT") #Assign treatments to examine
```

```{r}
X <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_genotype) #Treatments for Y
X <- X[, 14:198] #Retain data only
```

```{r}
known.plsda_CTvTT_25C <- plsda(X, Y, ncomp = 4) #Run PLSDA for desired contrast
VIP_CTvTT_25C <- as.data.frame(PLSDA.VIP(known.plsda_CTvTT_25C)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_CTvTT_25C
```

```{r}
clean_CTvTT_25C <- transMetabData_knownTimepoint %>%
  dplyr::filter(., treatment_genotype %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_CTvTT_25C <- cbind(clean_CTvTT_25C[, c(1:2, 6, 10)], 
                              log(clean_CTvTT_25C[, names(clean_CTvTT_25C) %in% VIP_CTvTT_25C$metabolite])) %>%
  pivot_longer(., cols = 5:76, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_CTvTT_25C #Confirm changes
```

```{r}
t.test_CTvTT_25C <-do(VIP_select_CTvTT_25C, tidy(t.test(.$log_norm_counts ~ .$treatment_genotype,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_CTvTT_25C$p.adj <- p.adjust(t.test_CTvTT_25C$p.value, method = c("fdr"), n = length(VIP_CTvTT_25C$metabolite)) #Adjust p value for the number of comparisons
t.test_CTvTT_25C
```

```{r}
t.test_CTvTT_25C %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

# Impact of genotype within each temperature

Above, I looked at the interaction between temperature and genotype across all the treatments. However, I have some temperature-specific expectations for each genotype. In cold temperatures, I expect having at least one C (cold-adapted) allele to be beneficial, and I expect having at least one T (warm-adapted) allele to be beneficial in warm temperatures. Since these alleles have differing roles in each tmeperature, and given that we know there are temperature-specific impacts on the metabolome, it may be hard to tease apart genotyping differences when considering both 5ºC and 25ºC. I'm going to look for genotype-specific impacts within each temperature.

## 5ºC

### PCA

#### All metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[-c(1:13)] #Data from after chronic thermal stress for 5ºC
```

```{r}
scaled.pca.5G <- prcomp((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.5G) #Standard deviations for each PC
```

```{r}
pcaPlot19 <- autoplot(scaled.pca.5G, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 1)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot19
ggsave("PCA/figures/all-data-PCA-5Genotype.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot20 <- autoplot(scaled.pca.5G, data = (transMetabData %>% filter(., day == "42")  %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot20
ggsave("PCA/figures/all-data-PCA-5PresenceC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot21 <- autoplot(scaled.pca.5G, data = (transMetabData %>% filter(., day == "42")%>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot21
ggsave("PCA/figures/all-data-PCA-5PresenceT.pdf", height = 8.5, width = 11)
```

#### Identified metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)] #Data from after chronic thermal stress for 5ºC
```

```{r}
scaled.pca.5G.id <- prcomp((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.5G.id) #Standard deviations for each PC
```

```{r}
pcaPlot22 <- autoplot(scaled.pca.5G.id, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 1)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot22
ggsave("PCA/figures/known-metabolites-PCA-5Genotype.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot23 <- autoplot(scaled.pca.5G.id, data = (transMetabData %>% filter(., day == "42")  %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot23
ggsave("PCA/figures/known-metabolites-PCA-5PresenceC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot24 <- autoplot(scaled.pca.5G.id, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot24
ggsave("PCA/figures/known-metabolites-PCA-5PresenceT.pdf", height = 8.5, width = 11)
```

##### Check outlier

```{r}
scaled.pca.5G.id.outlier <- prcomp((transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation without the potential outlier
summary(scaled.pca.5G.id.outlier) #Standard deviations for each PC
```

```{r}
pcaPlot25 <- autoplot(scaled.pca.5G.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", "steelblue1", plotColors[3]),
                    labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 1)),
                     labels = c("CC at 5ºC", "CT at 5ºC", "TT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot25
ggsave("PCA/figures/known-metabolites-PCA-5Genotype-no5075.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot26 <- autoplot(scaled.pca.5G.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")  %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("TT at 5ºC", "CC or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot26
ggsave("PCA/figures/known-metabolites-PCA-5PresenceC-no5075.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot27 <- autoplot(scaled.pca.5G.id.outlier, data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("lightblue2", plotColors[3]),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("lightblue2", plotColors[3]),
                    labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 5ºC", "TT or CT at 5ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot27
ggsave("PCA/figures/known-metabolites-PCA-5PresenceT-no5075.pdf", height = 8.5, width = 11)
```

### Conduct PERMANOVA

#### All metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[-c(1:13)] #Data from after chronic thermal stress for 5ºC
```

```{r}
dissim.5G <- vegdist(scale((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[-c(1:13)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.5GFull <- adonis2(dissim.5G ~ as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GFull #No significant impact of genotype
```

```{r}
disp.5GFull.gen <- betadisper(dissim.5G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GFull.gen) #PCoA
boxplot(disp.5GFull.gen) #Show distance to group centroids
anova(disp.5GFull.gen) #Variance is not different between groups.
```

```{r}
permanova.5GpresenceC <- adonis2(dissim.5G ~ as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")) %>% filter(., treatment == "05C"), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceC #No significant impact of genotype
```

```{r}
disp.5GpresenceC.gen <- betadisper(dissim.5G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GpresenceC.gen) #PCoA
boxplot(disp.5GpresenceC.gen) #Show distance to group centroids
anova(disp.5GpresenceC.gen) #Variance is not different between groups
```

```{r}
permanova.5GpresenceT <- adonis2(dissim.5G ~ as.factor(presenceT), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceT #No significant impact of genotype
```

```{r}
disp.5GpresenceT.gen <- betadisper(dissim.5G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GpresenceT.gen) #PCoA
boxplot(disp.5GpresenceT.gen) #Show distance to group centroids
anova(disp.5GpresenceT.gen) #Variance is not different between groups.
```

```{r}
permanova.5GHet <- adonis2(dissim.5G ~ as.factor(het), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GHet #No significant impact of genotype
```

```{r}
disp.5GHet.gen <- betadisper(dissim.5G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GHet.gen) #PCoA
boxplot(disp.5GHet.gen) #Show distance to group centroids
anova(disp.5GHet.gen) #Variance is not different between groups.
```

##### Save test output

```{r}
rbind(
  broom::tidy(permanova.5GFull) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.5GpresenceC) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.5GpresenceT) %>%
    mutate(genotype = "presenceT"), 
  broom::tidy(permanova.5GHet) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv(., "PCA/all-data-PERMANOVA-5Genotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.5GFull.gen)) %>%
    mutate(genotype = "full"),
  broom::tidy(anova(disp.5GpresenceC.gen)) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(anova(disp.5GpresenceC.gen)) %>%
    mutate(genotype = "presenceT"),
  broom::tidy(anova(disp.5GHet.gen)) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/all-data-PERMDISP-5Genotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

#### Identified metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)] #Data from after chronic thermal stress for 5ºC
```

```{r}
dissim.5G.id <- vegdist(scale((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.5GFull.id <- adonis2(dissim.5G.id ~ as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GFull.id #No significant impact of genotype
```

```{r}
disp.5GFull.gen.id <- betadisper(dissim.5G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GFull.gen.id) #PCoA
boxplot(disp.5GFull.gen.id) #Show distance to group centroids
anova(disp.5GFull.gen.id) #Variance is not different between groups.
```

```{r}
permanova.5GpresenceC.id <- adonis2(dissim.5G.id ~ as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")) %>% filter(., treatment == "05C"), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceC.id #No significant impact of genotype
```

```{r}
disp.5GPresenceC.gen.id <- betadisper(dissim.5G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GPresenceC.gen.id) #PCoA
boxplot(disp.5GPresenceC.gen.id) #Show distance to group centroids
anova(disp.5GPresenceC.gen.id) #Variance is not different between groups.
```

```{r}
permanova.5GpresenceT.id <- adonis2(dissim.5G.id ~ as.factor(presenceT), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceT.id #No significant impact of genotype
```

```{r}
disp.5GPresenceT.gen.id <- betadisper(dissim.5G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GPresenceT.gen.id) #PCoA
boxplot(disp.5GPresenceT.gen.id) #Show distance to group centroids
anova(disp.5GPresenceT.gen.id) #Variance is not different between groups.
```

```{r}
permanova.5GHet.id <- adonis2(dissim.5G.id ~ as.factor(het), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GHet.id #No significant impact of genotype
```

```{r}
disp.5GHet.gen.id <- betadisper(dissim.5G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GHet.gen.id) #PCoA
boxplot(disp.5GHet.gen.id) #Show distance to group centroids
anova(disp.5GHet.gen.id) #Variance is not different between groups.
```

##### Check outlier

I'm going to see if removal of 5-075 changes anything.

```{r}
dissim.5G.id.outlier <- vegdist(scale((transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.5GFull.id.outlier <- adonis2(dissim.5G.id.outlier ~ as.factor(final.genotype), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GFull.id.outlier #No significant impact of genotype
```

```{r}
disp.5GFull.gen.id.outlier <- betadisper(dissim.5G.id.outlier, group = as.factor((transMetabData  %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GFull.gen.id.outlier) #PCoA
boxplot(disp.5GFull.gen.id.outlier) #Show distance to group centroids
anova(disp.5GFull.gen.id.outlier) #Variance is not different between groups.
```

```{r}
permanova.5GpresenceC.id.outlier <- adonis2(dissim.5G.id.outlier ~ as.factor(presenceC), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42")) %>% filter(., treatment == "05C"), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceC.id.outlier #No significant impact of genotype
```

```{r}
disp.5GPresenceC.gen.id.outlier <- betadisper(dissim.5G.id.outlier, group = as.factor((transMetabData  %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GPresenceC.gen.id.outlier) #PCoA
boxplot(disp.5GPresenceC.gen.id.outlier) #Show distance to group centroids
anova(disp.5GPresenceC.gen.id.outlier) #Variance is not different between groups.
```

```{r}
permanova.5GpresenceT.id.outlier <- adonis2(dissim.5G.id.outlier ~ as.factor(presenceT), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GpresenceT.id.outlier #No significant impact of genotype
```

```{r}
disp.5GPresenceT.gen.id.outlier <- betadisper(dissim.5G.id.outlier, group = as.factor((transMetabData  %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GPresenceT.gen.id.outlier) #PCoA
boxplot(disp.5GPresenceT.gen.id.outlier) #Show distance to group centroids
anova(disp.5GPresenceT.gen.id.outlier) #Variance is not different between groups.
```

```{r}
permanova.5GHet.id.outlier <- adonis2(dissim.5G.id.outlier ~ as.factor(het), data = (transMetabData %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.5GHet.id.outlier #No significant impact of genotype
```

```{r}
disp.5GHet.gen.id.outlier <- betadisper(dissim.5G.id.outlier, group = as.factor((transMetabData  %>% filter(., crab.ID != "5-075") %>% filter(., day == "42") %>% filter(., treatment == "05C"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.5GHet.gen.id.outlier) #PCoA
boxplot(disp.5GHet.gen.id.outlier) #Show distance to group centroids
anova(disp.5GHet.gen.id.outlier) #Variance is not different between groups.
```

Removing the outlier doesn't change the statistical significance, so I'll include it in the results.

##### Save test output

```{r}
rbind(
  broom::tidy(permanova.5GFull.id) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.5GpresenceC.id) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.5GpresenceT.id) %>%
    mutate(genotype = "presenceT"), 
  broom::tidy(permanova.5GHet.id) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-5Genotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.5GFull.gen.id)) %>%
    mutate(genotype = "full"),
  broom::tidy(anova(disp.5GPresenceC.gen.id)) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(anova(disp.5GPresenceC.gen.id)) %>%
    mutate(genotype = "presenceT"),
  broom::tidy(anova(disp.5GHet.gen.id)) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-5Genotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

### PLS-DA

When doing the perMANOVAs, I explored the impact of various genotype factors on separation. Although none of them are significant, I am going to look at how presence of the C allele shapes separation in this supervised analysis. I expect that having at least one cold-adapted allele would improve performance in cold conditions.

#### Format data

```{r}
X_5Gen <- transMetabData %>% 
  filter(., day == "42") %>%
  filter(., treatment == "05C") %>%
  dplyr::select(c(14:198)) %>%
  log(.) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) #Assign known metabolite data to the X dataframe after log transformation. Replace any -Inf with NA, then replace NA with 0
Y_5Gen <- as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "05C"))$treatment_presenceC) #Assign treatment data to the Y dataframe
```

```{r}
nzv_indices <- nearZeroVar(X_5Gen) #Check to see if any variables have near-zero variance.
nzv_indices$Position #There aren't any columns with near-zero variance!
```

#### Tune PLS-DA model

```{r}
presenceC5.known.plsda <- plsda(X_5Gen, Y_5Gen, ncomp = 10) #Initially set a higher number for the number of components since we will be testing this
plotIndiv(presenceC5.known.plsda) #Preliminary PLS-DA
```

```{r}
perf.presenceC5.known.plsda <- perf(presenceC5.known.plsda, validation = "Mfold", folds = 4,  
                                    progressBar = TRUE, nrepeat = 50) 

plot(perf.presenceC5.known.plsda, sd = TRUE) #Overall, increasing the number of components decreases the error rate. There doesn't seem to be an exponential reduction of error rate after 2 components. All distances offers similarly low overall and balanced error rates (BER) when using 2 components
perf.presenceC5.known.plsda$choice.ncomp #Confirm information from plot.
```

#### Run PLS-DA

```{r}
presenceC5.known.plsda.tuned <- plsda(X_5Gen, Y_5Gen, ncomp = 2) #Run PLS-DA with the appropriate number of components to include based on tuning above.
plotIndiv(presenceC5.known.plsda.tuned, comp = c(1,2)) #Preliminary PLS-DA. I only have 4 TT crabs for this timepoint (not a mistake)
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-5presenceC-comp12.pdf", height = 8.5, width = 11)
plotIndiv(presenceC5.known.plsda.tuned, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", plotColors[3]),
          pch = as.factor(c(1, 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by C Allele Presence", X.label = "Component 1: 10% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 16% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Do not include a legend. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

#### Validate PLS-DA

```{r}
tune.splsda.5presenceC.known <- tune.splsda(X_5Gen, Y_5Gen, ncomp = 2, 
                                            validation = 'Mfold', folds = 4,
                                            measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.5presenceC.known) #Plot results
```

```{r}
error <- tune.splsda.5presenceC.known$error.rate #Save error rate
ncomp <- tune.splsda.5presenceC.known$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #1 component is the optimal number according to this analysis. However, looking at the error rate plot, error rate does go down as more features are added when using 2 components. I also need at least 2 components for this analysis.

select.keepX <- tune.splsda.5presenceC.known$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with this many features per component.
```

```{r}
final.splsda.5presenceC.known <- splsda(X_5Gen, Y_5Gen, ncomp = 2, keepX = select.keepX) #Use the number of components and metabolites to keep based on optimization from above
plotIndiv(final.splsda.5presenceC.known, comp = c(1,2))
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-5presenceC-comp12.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.5presenceC.known, comp = c(1,2), ind.names = FALSE,
          col = c("lightblue2", plotColors[3]),
          pch = as.factor(rep(c(1, 2), times = 1)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by C Allele Presence", X.label = "Component 1: 10% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 26% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

## 25ºC

### PCA

#### All metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[-c(1:13)] #Data from after chronic thermal stress for 5ºC
```

```{r}
scaled.pca.25G <- prcomp((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[-c(1:13)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.25G) #Standard deviations for each PC
```

```{r}
pcaPlot28 <- autoplot(scaled.pca.25G, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 1)),
                     labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot28
ggsave("PCA/figures/all-data-PCA-25Genotype.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot29 <- autoplot(scaled.pca.25G, data = (transMetabData %>% filter(., day == "42")  %>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", plotColors[1]),
                     labels = c("TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", plotColors[1]),
                    labels = c("TT at 25ºC", "CC or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("TT at 25ºC", "CC or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot29
ggsave("PCA/figures/all-data-PCA-25PresenceC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot30 <- autoplot(scaled.pca.25G, data = (transMetabData %>% filter(., day == "42")%>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", plotColors[1]),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", plotColors[1]),
                    labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot30
ggsave("PCA/figures/all-data-PCA-25PresenceT.pdf", height = 8.5, width = 11)
```

#### Identified metabolites

```{r}
(transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[c(14:198)] #Data from after chronic thermal stress for 5ºC
```

```{r}
scaled.pca.25G.id <- prcomp((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[c(14:198)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.25G.id) #Standard deviations for each PC
```

```{r}
pcaPlot31 <- autoplot(scaled.pca.25G.id, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_genotype", shape = "treatment_genotype", 
                      frame = TRUE, frame.colour = "treatment_genotype", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", "firebrick1", plotColors[1]),
                     labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", "firebrick1", plotColors[1]),
                    labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2, 0), times = 1)),
                     labels = c("CC at 25ºC", "CT at 25ºC", "TT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot31
ggsave("PCA/figures/known-metabolites-PCA-25Genotype.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot32 <- autoplot(scaled.pca.25G.id, data = (transMetabData %>% filter(., day == "42")  %>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceC", shape = "treatment_presenceC", 
                      frame = TRUE, frame.colour = "treatment_presenceC", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", plotColors[1]),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", plotColors[1]),
                    labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot32
ggsave("PCA/figures/known-metabolites-PCA-25PresenceC.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot33 <- autoplot(scaled.pca.25G.id, data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), x = 1, y = 2,
                      size = 4, alpha = 0.8, colour = "treatment_presenceT", shape = "treatment_presenceT", 
                      frame = TRUE, frame.colour = "treatment_presenceT", loadings = FALSE) +
  scale_color_manual(values = c("salmon2", plotColors[1]),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  scale_fill_manual(values = c("salmon2", plotColors[1]),
                    labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                    name = "Treatment x Genotype") +
  scale_shape_manual(values = c(rep(c(1, 2), times = 1)),
                     labels = c("CC at 25ºC", "TT or CT at 25ºC"),
                     name = "Treatment x Genotype") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot33
ggsave("PCA/figures/known-metabolites-PCA-25PresenceT.pdf", height = 8.5, width = 11)
```

### Conduct PERMANOVA

#### All metabolites

```{r}
dissim.25G <- vegdist(scale((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[-c(1:13)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.25GFull <- adonis2(dissim.25G ~ as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GFull #No significant impact of genotype
```

```{r}
disp.25GFull.gen <- betadisper(dissim.25G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GFull.gen) #PCoA
boxplot(disp.25GFull.gen) #Show distance to group centroids
anova(disp.25GFull.gen) #Variance is not different between groups.
```

```{r}
permanova.25GpresenceC <- adonis2(dissim.25G ~ as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")) %>% filter(., treatment == "25C"), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GpresenceC #No significant impact of genotype
```

```{r}
disp.25GPresenceC.gen <- betadisper(dissim.25G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GPresenceC.gen) #PCoA
boxplot(disp.25GPresenceC.gen) #Show distance to group centroids
anova(disp.25GPresenceC.gen) #Variance is not different between groups.
```

```{r}
permanova.25GpresenceT <- adonis2(dissim.25G ~ as.factor(presenceT), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GpresenceT #No significant impact of genotype
```

```{r}
disp.25GPresenceT.gen <- betadisper(dissim.25G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GPresenceT.gen) #PCoA
boxplot(disp.25GPresenceT.gen) #Show distance to group centroids
anova(disp.25GPresenceT.gen) #Significant difference in variance! Will likely be non-significant after multiple test correction
```

```{r}
permanova.25GHet <- adonis2(dissim.25G ~ as.factor(het), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GHet #No significant impact of genotype
```

```{r}
disp.25GHet.gen <- betadisper(dissim.25G, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GHet.gen) #PCoA
boxplot(disp.25GHet.gen) #Show distance to group centroids
anova(disp.25GHet.gen) #Variance is not different between groups.
```

##### Save test output

```{r}
rbind(
  broom::tidy(permanova.25GFull) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.25GpresenceC) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.25GpresenceT) %>%
    mutate(genotype = "presenceT"), 
  broom::tidy(permanova.25GHet) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv(., "PCA/all-data-PERMANOVA-25Genotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.25GFull.gen)) %>%
    mutate(genotype = "full"),
  broom::tidy(anova(disp.25GPresenceC.gen)) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(anova(disp.25GPresenceT.gen)) %>%
    mutate(genotype = "presenceT"),
  broom::tidy(anova(disp.25GHet.gen)) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/all-data-PERMDISP-25Genotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

#### Identified metabolites

```{r}
dissim.25G.id <- vegdist(scale((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))[c(14:198)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.25GFull.id <- adonis2(dissim.25G.id ~ as.factor(final.genotype), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GFull.id #No significant impact of genotype
```

```{r}
disp.25GFull.gen.id <- betadisper(dissim.25G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$final.genotype), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GFull.gen.id) #PCoA
boxplot(disp.25GFull.gen.id) #Show distance to group centroids
anova(disp.25GFull.gen.id) #Variance is not different between groups.
```

```{r}
permanova.25GpresenceC.id <- adonis2(dissim.25G.id ~ as.factor(presenceC), data = (transMetabData %>% filter(., day == "42")) %>% filter(., treatment == "25C"), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GpresenceC.id #No significant impact of genotype
```

```{r}
disp.25GPresenceC.gen.id <- betadisper(dissim.25G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$presenceC), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GPresenceC.gen.id) #PCoA
boxplot(disp.25GPresenceC.gen.id) #Show distance to group centroids
anova(disp.25GPresenceC.gen.id) #Variance is not different between groups.
```

```{r}
permanova.25GpresenceT.id <- adonis2(dissim.25G.id ~ as.factor(presenceT), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GpresenceT.id #No significant impact of genotype
```

```{r}
disp.25GPresenceT.gen.id <- betadisper(dissim.25G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$presenceT), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GPresenceT.gen.id) #PCoA
boxplot(disp.25GPresenceT.gen.id) #Show distance to group centroids
anova(disp.25GPresenceT.gen.id) #Variance is different between groups! Likely will be non-significant after multiple test correction
```

```{r}
permanova.25GHet.id <- adonis2(dissim.25G.id ~ as.factor(het), data = (transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C")), method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.25GHet.id #No significant impact of genotype
```

```{r}
disp.25GHet.gen.id <- betadisper(dissim.25G.id, group = as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$het), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.25GHet.gen.id) #PCoA
boxplot(disp.25GHet.gen.id) #Show distance to group centroids
anova(disp.25GHet.gen.id) #Variance is different between groups! Likely will be non-significant after multiple test correction
```

##### Save test output

```{r}
rbind(
  broom::tidy(permanova.25GFull.id) %>%
    mutate(genotype = "full"),
  broom::tidy(permanova.25GpresenceC.id) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(permanova.25GpresenceT.id) %>%
    mutate(genotype = "presenceT"), 
  broom::tidy(permanova.25GHet.id) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-25Genotype.csv", quote = FALSE) #Save PERMANOVA output
```

```{r}
rbind(
  broom::tidy(anova(disp.25GFull.gen.id)) %>%
    mutate(genotype = "full"),
  broom::tidy(anova(disp.25GPresenceC.gen.id)) %>%
    mutate(genotype = "presenceC"),
  broom::tidy(anova(disp.25GPresenceT.gen.id)) %>%
    mutate(genotype = "presenceT"),
  broom::tidy(anova(disp.25GHet.gen.id)) %>%
    mutate(genotype = "het")
) %>%
  mutate(., p.adj = p.adjust(p.value, method = "bonferroni")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-25Genotype.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

Presence of the T allele's impact on dispersion is marginally significant after multiple correction testing! That may justify keeping genotype in the mix for the PLS-DA.

### PLS-DA

I'm going to look at the impact of the T allele on the PLS-DA based on the marginally significant disperson result.

#### Format data

```{r}
X_25Gen <- transMetabData %>% 
  filter(., day == "42") %>%
  filter(., treatment == "25C") %>%
  dplyr::select(c(14:198)) %>%
  log(.) %>% 
  mutate_if(is.numeric, list(~na_if(., -Inf))) %>%
  replace(is.na(.), 0) #Assign known metabolite data to the X dataframe after log transformation. Replace any -Inf with NA, then replace NA with 0
Y_25Gen <- as.factor((transMetabData %>% filter(., day == "42") %>% filter(., treatment == "25C"))$treatment_presenceT) #Assign treatment data to the Y dataframe
```

```{r}
nzv_indices <- nearZeroVar(X_5Gen) #Check to see if any variables have near-zero variance.
nzv_indices$Position #There aren't any columns with near-zero variance!
```

#### Tune PLS-DA model

```{r}
presenceT25.known.plsda <- plsda(X_25Gen, Y_25Gen, ncomp = 10) #Initially set a higher number for the number of components since we will be testing this
plotIndiv(presenceT25.known.plsda) #Preliminary PLS-DA
```

```{r}
perf.presenceT25.known.plsda <- perf(presenceT25.known.plsda, validation = "Mfold", folds = 4,  
                                     progressBar = TRUE, nrepeat = 50) 

plot(perf.presenceT25.known.plsda, sd = TRUE) #The lowest error rate is actually achieved at 1 component, but I need 2 components to do the analysis.
perf.presenceT25.known.plsda$choice.ncomp #Confirm information from plot.
```

#### Run PLS-DA

```{r}
presenceT25.known.plsda.tuned <- plsda(X_25Gen, Y_25Gen, ncomp = 2) #Run PLS-DA with the appropriate number of components to include based on tuning above.
plotIndiv(presenceT25.known.plsda.tuned, comp = c(1,2)) #Preliminary PLS-DA. I only have 5 CC crabs for this timepoint (not a mistake)
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-25presenceT-comp12.pdf", height = 8.5, width = 11)
plotIndiv(presenceT25.known.plsda.tuned, comp = c(1,2), ind.names = FALSE,
          col = c("salmon2", plotColors[1]),
          pch = as.factor(c(1, 2)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = FALSE,
          title = "PLS-DA by T Allele Presence", X.label = "Component 1: 11% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 9% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Do not include a legend. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

#### Validate PLS-DA

```{r}
tune.splsda.25presenceT.known <- tune.splsda(X_25Gen, Y_25Gen, ncomp = 2, 
                                             validation = 'Mfold', folds = 4,
                                             measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.25presenceT.known) #Plot results
```

```{r}
error <- tune.splsda.25presenceT.known$error.rate #Save error rate
ncomp <- tune.splsda.25presenceT.known$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #1 component is the optimal number according to this analysis. However, looking at the error rate plot, error rate does go down as more features are added when using 2 components. I also need at least 2 components for this analysis.

select.keepX <- tune.splsda.25presenceT.known$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with this many features per component.
```

```{r}
final.splsda.25presenceT.known <- splsda(X_25Gen, Y_25Gen, ncomp = 2, keepX = select.keepX) #Use the number of components and metabolites to keep based on optimization from above
plotIndiv(final.splsda.25presenceT.known, comp = c(1,2))
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-25presenceT-comp12.pdf", height = 8.5, width = 11)
plotIndiv(final.splsda.25presenceT.known, comp = c(1,2), ind.names = FALSE,
          col = c("salmon2", plotColors[1]),
          pch = as.factor(rep(c(1, 2), times = 1)), cex = 3, point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          title = "sPLS-DA by T Allele Presence", X.label = "Component 1: 7% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 11% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```







# DON'T LOOK BELOW HERE IT'S EMBARASSING #

# CODE IN PROGRESS BELOW #

# Pathway analysis

The final step in this workflow is to understand the various metabolic pathways represented in each module eigengene. To do this, I will conduct a manual pathway analysis and identify VIP present in each significant cluster, and also use MetaboAnalyst.

```{bash}
mkdir metaboanalyst
mkdir metaboanalyst/figures
```

## Manual pathway analysis

There are a few pathways of interest based on my experiment. Since thermal tolerance in marine invertebrates is linked to oxygen availability, the tradeoff between aerobic and anaerobic respiration can tell me if a crab was tipping into a pejus or critical range. Additionally, previous research shows that magnesium can block calcium channels in colder temperatures, reducing mobility. I wanted to see if any metabolites associated with calcium signaling had different abundances according to temperature, which may provide additional evidence for magnesium blockage.

### Aerobic respiration

#### Glycolysis

1. Glucose: Glucose is the starting substrate for glycolysis. It is phosphorylated to form glucose-6-phosphate by hexokinase or glucokinase.
2. Glucose-1-phosphate: Glucose-1-phosphate is an intermediate in the breakdown of glycogen, which can be converted into glucose-6-phosphate before entering glycolysis.
3. Glucose-6-phosphate: It is the first intermediate in glycolysis, formed from glucose by hexokinase. It is subsequently converted into fructose-6-phosphate.
4. Fructose-6-phosphate: It is an intermediate in glycolysis, formed from glucose-6-phosphate by phosphoglucose isomerase. It is subsequently phosphorylated to fructose-1,6-bisphosphate.
5. Fructose-1-phosphate: Fructose-1-phosphate is an intermediate in fructose metabolism. It is converted into glyceraldehyde-3-phosphate (an intermediate in glycolysis) and dihydroxyacetone phosphate by aldolase.
6. 3-Phosphoglycerate: It is an intermediate in glycolysis, formed from 1,3-bisphosphoglycerate by phosphoglycerate kinase. It is subsequently converted into 2-phosphoglycerate.

```{r}
glycolysis_facets <- c("1" = "glucose",
                       "2" = "glucose-1-phosphate",
                       "3" = "glucose-6-phosphate",
                       "4" = "fructose-6-phosphate",
                       "5" = "fructose-1-phosphate",
                       "6" = "3-phosphoglycerate") #Assign labels for facets
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, 
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`)) %>%
  pivot_longer(., cols = c(5:10), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glucose " ~ 1,
                              metabolite == "glucose-1-phosphate" ~ 2,
                              metabolite == "glucose-6-phosphate" ~ 3,
                              metabolite == "fructose-6-phosphate " ~ 4,
                              metabolite == "fructose-1-phosphate" ~ 5,
                              metabolite == "3-phosphoglycerate" ~ 6)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = glycolysis_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/glycolysis-plot.pdf", height = 8.5, width = 11)
```

#### TCA cycle

- Citric acid: Also known as citrate, it is the first intermediate in the TCA cycle. It is formed from acetyl-CoA and oxaloacetate and undergoes a series of reactions to regenerate oxaloacetate, producing NADH, FADH2, and GTP in the process.
- Alpha-ketoglutarate: Also known as 2-oxoglutarate, it is an intermediate in the TCA cycle. It is formed from isocitrate by isocitrate dehydrogenase and is subsequently converted into succinyl-CoA.
- Fumaric acid: Fumarate is an intermediate in the TCA cycle, formed by the hydration of fumarate catalyzed by fumarase. It is subsequently converted into malate.
- Malic acid: Malate is an intermediate in the TCA cycle, formed from fumarate by fumarase. It is subsequently converted into oxaloacetate.

```{r}
TCA_facets <- c("1" = "citric acid",
                "2" = "alpha-ketoglutarate",
                "3" = "fumaric acid",
                "4" = "malic acid") #Assign labels for facets
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, 
                     `citric acid`, `alpha-ketoglutarate`, `fumaric acid`, `malic acid`)) %>%
  pivot_longer(., cols = c(5:8), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "citric acid" ~ 1,
                              metabolite == "alpha-ketoglutarate" ~ 2,
                              metabolite == "fumaric acid" ~ 3,
                              metabolite == "malic acid" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = TCA_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/TCA-plot.pdf", height = 8.5, width = 11)
```

### Anaerobic respiration

1. Glucose: Glucose is the starting substrate for glycolysis, the metabolic pathway that generates pyruvate, the precursor of lactic acid.
2. Glucose-1-phosphate: Glucose-1-phosphate is an intermediate in the breakdown of glycogen, which can be converted into glucose-6-phosphate before entering glycolysis.
3. Glucose-6-phosphate: It is the first intermediate in glycolysis, formed from glucose by hexokinase. It is subsequently converted into fructose-6-phosphate.
4. Fructose-6-phosphate: It is an intermediate in glycolysis, formed from glucose-6-phosphate by phosphoglucose isomerase. It is subsequently phosphorylated to fructose-1,6-bisphosphate.
5. Fructose-1-phosphate: Fructose-1-phosphate is an intermediate in fructose metabolism. It is converted into glyceraldehyde-3-phosphate (an intermediate in glycolysis) and dihydroxyacetone phosphate by aldolase.
6. 3-Phosphoglycerate: It is an intermediate in glycolysis, formed from 1,3-bisphosphoglycerate by phosphoglycerate kinase. It is subsequently converted into 2-phosphoglycerate.
7. Lactic acid: The end product of lactic acid fermentation, formed from the reduction of pyruvate.

```{r}
anaerobic_facets <- c(glycolysis_facets,
                      "7" = "lactic acid") #Make facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`, `lactic acid`)) %>%
  pivot_longer(., cols = c(5:11), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glucose " ~ 1,
                              metabolite == "glucose-1-phosphate" ~ 2,
                              metabolite == "glucose-6-phosphate" ~ 3,
                              metabolite == "fructose-6-phosphate " ~ 4,
                              metabolite == "fructose-1-phosphate" ~ 5,
                              metabolite == "3-phosphoglycerate" ~ 6,
                              metabolite == "lactic acid" ~ 7)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = anaerobic_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/anaerobic-metabolism-plot.pdf", height = 8.5, width = 11)
```

### Combined respiration plot

I'm going to combine metabolite abundance information from aerobic and anaerobic respiration pathways using InDesign. To do this, I need individual plots for each metabolite that are the same size.

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`,
                     `lactic acid`,
                     `citric acid`, `alpha-ketoglutarate`, `fumaric acid`, `malic acid`)) %>%
  pivot_longer(., cols = c(5:15), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/combined-respiration-inputs.pdf", height = 8.5, width = 11)
```

```{r}
respirationMetabolites <- c("glucose ", "glucose-1-phosphate", "glucose-6-phosphate", "fructose-6-phosphate ", "fructose-1-phosphate", "3-phosphoglycerate",
                            "lactic acid",
                            "citric acid", "alpha-ketoglutarate", "fumaric acid", "malic acid") #Make a vector with metabolites of interest
```

```{r}
overallVIP_respirationMetabolites <- significantVIP %>%
  filter(., metabolite %in% respirationMetabolites) #Identify overall VIP associated with pathway
overallVIP_respirationMetabolites
```

```{r}
write.table(overallVIP_respirationMetabolites, "metaboanalyst/combined-respiration-overall-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
VIP_respirationMetabolites <- rbind(t.test_13v30_day3 %>%
                                      mutate(., comparison = "13v30_day3") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v5_day3 %>%
                                      mutate(., comparison = "13v5_day3") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v30_day22 %>%
                                      mutate(., comparison = "13v30_day22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v5_day22 %>%
                                      mutate(., comparison = "13v5_day22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_5_day3v22 %>%
                                      mutate(., comparison = "5_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13_day3v22 %>%
                                      mutate(., comparison = "13_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_30_day3v22 %>%
                                      mutate(., comparison = "30_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites))
VIP_respirationMetabolites
```

```{r}
write.table(VIP_respirationMetabolites, "metaboanalyst/combined-respiration-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

### Calcium signaling

- UDP-N-acetylglucosamine: Calcium signaling can influence the activity of enzymes involved in the synthesis of UDP-N-acetylglucosamine. Blocking calcium channels with Mg2+ could potentially reduce the production of UDP-N-acetylglucosamine, leading to decreased concentrations.
- Tyrosine: Calcium signaling pathways can regulate the activity of enzymes involved in tyrosine metabolism. Blocking calcium channels may disrupt these pathways, potentially altering the concentration of tyrosine.
- Tryptophan: Similar to tyrosine, tryptophan metabolism can be influenced by calcium signaling. Blocking calcium channels may affect tryptophan metabolism and alter its concentration.
- Threonine: Threonine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect threonine metabolism, potentially leading to alterations in its concentration.
Serotonin: Calcium signaling pathways play a crucial role in the release and regulation of serotonin in smooth muscle cells. Blocking calcium channels with Mg2+ could lead to decreased serotonin release and lower concentrations.
- Serine: Similar to threonine, serine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect serine metabolism, potentially leading to alterations in its concentration.
- Phosphoinositol: Phosphoinositol lipids are components of cell membranes and can serve as precursors for signaling molecules such as inositol trisphosphate (IP3), which plays a key role in intracellular calcium release from the endoplasmic reticulum.
- Citric acid: Citric acid is an intermediate in the citric acid cycle, which can be influenced by calcium signaling pathways. Blocking calcium channels may disrupt the citric acid cycle, potentially leading to alterations in citric acid concentration.
- Glutathione: Calcium signaling pathways can regulate the synthesis and metabolism of glutathione. Blocking calcium channels may disrupt glutathione metabolism, potentially altering its concentration.
- Glutamine: Glutamine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect glutamine metabolism, potentially leading to alterations in its concentration.
- Glutamate: Glutamate is an excitatory neurotransmitter that can be influenced by calcium signaling pathways. Blocking calcium channels may disrupt glutamate release and metabolism, potentially altering its concentration.

```{r}
calciumSignalingMetabolites <- c("UDP-N-acetylglucosamine ", "tyrosine", "tryptophan ", "threonine", "serine", "phosphoinositol", "citric acid", "glutathione ", "glutamine", "glutamate ") #Create a vector with metabolites of interest
```

```{r}
overallVIP_calciumSignalingMetabolites <- significantVIP %>%
  filter(., metabolite %in% calciumSignalingMetabolites) #Identify overall VIP associated with pathway
overallVIP_calciumSignalingMetabolites
```

```{r}
write.table(overallVIP_calciumSignalingMetabolites, "metaboanalyst/calcium-signaling-overall-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
VIP_calciumSignalingMetabolites <- rbind(t.test_13v30_day3 %>%
                                           mutate(., comparison = "13v30_day3") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v5_day3 %>%
                                           mutate(., comparison = "13v5_day3") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v30_day22 %>%
                                           mutate(., comparison = "13v30_day22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v5_day22 %>%
                                           mutate(., comparison = "13v5_day22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_5_day3v22 %>%
                                           mutate(., comparison = "5_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13_day3v22 %>%
                                           mutate(., comparison = "13_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_30_day3v22 %>%
                                           mutate(., comparison = "30_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites))
VIP_calciumSignalingMetabolites
```

```{r}
write.table(VIP_calciumSignalingMetabolites, "metaboanalyst/calcium-signaling-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
calciumSignaling_facets <- c("1" = "glutamine",
                             "2" = "glutathione ",
                             "3" = "threonine",
                             "4" = "tryptophan ",
                             "5" = "UDP-N-acetylglucosamine ",
                             "6" = "serine",
                             "7" = "citric acid",
                             "8" = "glutamate ",
                             "9" = "phosphoinositol",  
                             "10" = "tyrosine") #Create facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `UDP-N-acetylglucosamine `, tyrosine, `tryptophan `, threonine, serine, phosphoinositol, `citric acid`, `glutathione `, glutamine, `glutamate `)) %>%
  pivot_longer(., cols = c(5:14), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glutamine" ~ 1,
                              metabolite == "glutathione " ~ 2,
                              metabolite == "threonine" ~ 3,
                              metabolite == "tryptophan " ~ 4,
                              metabolite == "UDP-N-acetylglucosamine " ~ 5,
                              metabolite == "serine" ~ 6,
                              metabolite == "citric acid" ~7,
                              metabolite == "glutamate " ~ 8,
                              metabolite == "phosphoinositol" ~ 9,
                              metabolite == "tyrosine" ~ 10)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free", labeller = labeller(order = calciumSignaling_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/calcium-signaling-plot.pdf", height = 8.5, width = 11)
```

### Cold protectors

```{r}
coldProtectionMetabolites <- c("alanine", "ornithine ", "proline") #Create a vector with metabolites of interest
```

```{r}
overallVIP_coldProtectionMetabolites <- significantVIP %>%
  filter(., metabolite %in% coldProtectionMetabolites) #Identify overall VIP associated with pathway
overallVIP_coldProtectionMetabolites
```

```{r}
VIP_coldProtectionMetabolites <- rbind(t.test_13v30_day3 %>%
                                         mutate(., comparison = "13v30_day3") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v5_day3 %>%
                                         mutate(., comparison = "13v5_day3") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v30_day22 %>%
                                         mutate(., comparison = "13v30_day22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v5_day22 %>%
                                         mutate(., comparison = "13v5_day22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_5_day3v22 %>%
                                         mutate(., comparison = "5_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13_day3v22 %>%
                                         mutate(., comparison = "13_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_30_day3v22 %>%
                                         mutate(., comparison = "30_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites))
VIP_coldProtectionMetabolites
```

```{r}
write.table(VIP_coldProtectionMetabolites, "metaboanalyst/cold-protection-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
coldProtection_facets <- c("1" = "ornithine ",
                           "2" = "proline",
                           "3" = "alanine") #Create facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     alanine, `ornithine `, proline)) %>%
  pivot_longer(., cols = c(5:7), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "ornithine " ~ 1,
                              metabolite == "proline" ~ 2,
                              metabolite == "alanine" ~ 3)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free", labeller = labeller(order = coldProtection_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/cold-protection-plot.pdf", height = 8.5, width = 11)
```

## VIP "enrichment" analysis

The second step of this workflow is to analyze the abundance patterns and associated functions/pathways of significant VIP.

### Pairwise VIP

#### MetaboAnalyst

I'll conduct enrichment tests using pairwise VIP from my various contrasts, since that will shed more light on short- and long-term acclimation processes to various temperatures.

```{r}
t.test_30v5_day3 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/30v5-day3_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_13v30_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/13v30-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_13v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/13v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_30v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("../output/05-metabolomics-analysis/metaboanalyst/30v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_30v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("../output/05-metabolomics-analysis/metaboanalyst/unique-30v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_5_day3v22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/5-day3v22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

Within MetaboAnalyst, I did the following:

- Uploaded each list of metabolite compound names for significant VIP in the Pathway Analysis module
- Matched to KEGG pathways from *Drosophila melanogaster*

##### Day 3: 30ºC vs. 5ºC

```{r}
day3_30v5_pathwayResults <- read.csv("metaboanalyst/30v5-day3_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day3_30v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day3_30v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day3-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3_30v5_pathwayResults <- day3_30v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day3_30v5_pathwayResults) #Confirm formatting
```

```{r}
top_day3_30v5_pathwayResults %>%
  mutate(., order = rev(1:12)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_day3_30v5_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day3-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 13ºC vs. 30ºC

```{r}
significantVIP_13v30_day22 <- t.test_13v30_day22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_13v30_day22) #Confirm dataframe creation
nrow(significantVIP_13v30_day22)
```

```{r}
#pdf("metaboanalyst/figures/13v30_day22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "30ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/13v30_day22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "30ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `30_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "30")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[1]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v30-day22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `30_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "30")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[1]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v30-day22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day22_13v30_pathwayResults <- read.csv("metaboanalyst/13v30-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_13v30_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_13v30_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults <- day22_13v30_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_13v30_pathwayResults) #Confirm formatting
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 13ºC vs. 5ºC

```{r}
significantVIP_13v5_day22 <- t.test_13v5_day22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_13v5_day22) #Confirm dataframe creation
nrow(significantVIP_13v5_day22)
```

```{r}
#pdf("metaboanalyst/figures/13v5_day22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/13v5_day22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "5")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[3]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v5-day22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "5")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[3]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v5-day22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day22_13v5_pathwayResults <- read.csv("metaboanalyst/13v5-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_13v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_13v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 2.5) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults <- day22_13v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_13v5_pathwayResults) #Confirm formatting
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 30ºC vs. 5ºC

```{r}
day22_30v5_pathwayResults <- read.csv("metaboanalyst/unique-30v5-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_30v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_30v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_30v5_pathwayResults <- day22_30v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_30v5_pathwayResults) #Confirm formatting
```

```{r}
top_day22_30v5_pathwayResults %>%
  mutate(., order = rev(1:10)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_day22_30v5_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

##### 5ºC: Day 3 vs. Day 22

```{r}
significantVIP_5_day3v22 <- t.test_5_day3v22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_5_day3v22) #Confirm dataframe creation
nrow(significantVIP_5_day3v22)
```

```{r}
#pdf("metaboanalyst/figures/5_day3v22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
            filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "5ºC at Day 3" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/5_day3v22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
            filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "5ºC at Day 3" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
  filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `5_3` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "3",
                               logDiff < 0 ~ "22")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c("black", "purple"), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-5-day3v22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
  filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `5_3` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "3",
                               logDiff < 0 ~ "22")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c("black", "purple"), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-5-day3v22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day3v22_5_pathwayResults <- read.csv("metaboanalyst/5-day3v22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day3v22_5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day3v22_5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 3) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults <- day3v22_5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day3v22_5_pathwayResults) #Confirm formatting
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

#### Identify pairwise VIP in each module

I'm only interested in the following modules significantly correlated with specific treatment conditions:

- Module 0: 5 and 30 at day 22
- Module 2: 5 at day 3 and 30 at day 22
- Module 3: 5 and 30 at day 22
- Module 4: 30 at a day 22
- Module 5: 5 at day 22, 30 at day 22, 30 at day 3
- Module 6: 13 at day 22
- Module 7: 13 and 5 at day 22

```{r}
VIP_module0 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite)) %>% 
  mutate(module = "module0") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module0
```

```{r}
VIP_module2 <- rbind(t.test_13v5_day3 %>%
                       mutate(., comparison = "13v5_day3") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite))  %>% 
  mutate(module = "module2") #Identify VIP present in module 2 across all pairwise comparisons
VIP_module2
```

```{r}
VIP_module3 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite))  %>% 
  mutate(module = "module3") #Identify VIP present in module 3 across all pairwise comparisons
VIP_module3
```

```{r}
VIP_module4 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module4_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module4_metabolites$Metabolite))  %>% 
  mutate(module = "module4") #Identify VIP present in module 4 across all pairwise comparisons
VIP_module4
```

```{r}
VIP_module5 <- rbind(t.test_13v30_day3 %>%
                       mutate(., comparison = "13v30_day3") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite))  %>% 
  mutate(module = "module5") #Identify VIP present in module 5 across all pairwise comparisons
VIP_module5
```

```{r}
VIP_module6 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite),
                     t.test_13_day3v22 %>%
                       mutate(., comparison = "13_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite))  %>% 
  mutate(module = "module6") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module6
```

```{r}
VIP_module7 <- rbind(t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module7_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module7_metabolites$Metabolite))  %>% 
  mutate(module = "module7") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module7
```

```{r}
VIP_allModules <- rbind(VIP_module0,
                        VIP_module2,
                        VIP_module3,
                        VIP_module4,
                        VIP_module5,
                        VIP_module6,
                        VIP_module7) #Combine all VIP lists
head(VIP_allModules)
```

```{r}
write.table(VIP_allModules, "metaboanalyst/all-module-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Identify pairwise VIP that overlap with significant pathways

```{r}
VIP_valineBiosynthesis <- rbind(t.test_13v30_day3 %>%
                                  mutate(., comparison = "13v30_day3") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v5_day3 %>%
                                  mutate(., comparison = "13v5_day3") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v30_day22 %>%
                                  mutate(., comparison = "13v30_day22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v5_day22 %>%
                                  mutate(., comparison = "13v5_day22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_5_day3v22 %>%
                                  mutate(., comparison = "5_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13_day3v22 %>%
                                  mutate(., comparison = "13_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_30_day3v22 %>%
                                  mutate(., comparison = "30_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis)) %>%
  mutate(., pathway = "valineBiosynthesis")
VIP_valineBiosynthesis
```

```{r}
VIP_arginineBiosynthesis <- rbind(t.test_13v30_day3 %>%
                                    mutate(., comparison = "13v30_day3") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v5_day3 %>%
                                    mutate(., comparison = "13v5_day3") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v30_day22 %>%
                                    mutate(., comparison = "13v30_day22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v5_day22 %>%
                                    mutate(., comparison = "13v5_day22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_5_day3v22 %>%
                                    mutate(., comparison = "5_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13_day3v22 %>%
                                    mutate(., comparison = "13_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_30_day3v22 %>%
                                    mutate(., comparison = "30_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis)) %>%
  mutate(., pathway = "arginineBiosynthesis")
VIP_arginineBiosynthesis
```
```{r}
VIP_glutathioneMetabolism <- rbind(t.test_13v30_day3 %>%
                                     mutate(., comparison = "13v30_day3") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v5_day3 %>%
                                     mutate(., comparison = "13v5_day3") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v30_day22 %>%
                                     mutate(., comparison = "13v30_day22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v5_day22 %>%
                                     mutate(., comparison = "13v5_day22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_5_day3v22 %>%
                                     mutate(., comparison = "5_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13_day3v22 %>%
                                     mutate(., comparison = "13_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_30_day3v22 %>%
                                     mutate(., comparison = "30_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism)) %>%
  mutate(., pathway = "glutathioneMetabolism")
VIP_glutathioneMetabolism
```
```{r}
VIP_alanineMetabolism <- rbind(t.test_13v30_day3 %>%
                                 mutate(., comparison = "13v30_day3") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v5_day3 %>%
                                 mutate(., comparison = "13v5_day3") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v30_day22 %>%
                                 mutate(., comparison = "13v30_day22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v5_day22 %>%
                                 mutate(., comparison = "13v5_day22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_5_day3v22 %>%
                                 mutate(., comparison = "5_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13_day3v22 %>%
                                 mutate(., comparison = "13_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_30_day3v22 %>%
                                 mutate(., comparison = "30_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism)) %>%
  mutate(., pathway = "alanineMetabolism")
VIP_alanineMetabolism
```

```{r}
VIP_allPathways <- rbind(VIP_valineBiosynthesis,
                         VIP_arginineBiosynthesis,
                         VIP_glutathioneMetabolism,
                         VIP_alanineMetabolism)
VIP_allPathways
```

```{r}
write.table(VIP_allPathways, "metaboanalyst/significant-pathways-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Identify pairwise VIP that overlap with VIP from significant pathways for each module

```{r}
module0_pairwiseVIPpathway <- rbind(VIP_module0 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module0") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module0_pairwiseVIPpathway
```

```{r}
module2_pairwiseVIPpathway <- rbind(VIP_module2 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module2") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module2_pairwiseVIPpathway
```

```{r}
module3_pairwiseVIPpathway <- rbind(VIP_module3 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module3") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module3_pairwiseVIPpathway
```
```{r}
module4_pairwiseVIPpathway <- rbind(VIP_module4 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module4") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module4_pairwiseVIPpathway
```

```{r}
module5_pairwiseVIPpathway <- rbind(VIP_module5 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module5") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module5_pairwiseVIPpathway
```

```{r}
module6_pairwiseVIPpathway <- rbind(VIP_module6 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module6") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module6_pairwiseVIPpathway
```

```{r}
module7_pairwiseVIPpathway <- rbind(VIP_module7 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module7") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module7_pairwiseVIPpathway
```

```{r}
pairwiseVIPpathway_allModules <- rbind(module0_pairwiseVIPpathway,
                                       module2_pairwiseVIPpathway,
                                       module3_pairwiseVIPpathway,
                                       module4_pairwiseVIPpathway,
                                       module5_pairwiseVIPpathway,
                                       module6_pairwiseVIPpathway,
                                       module7_pairwiseVIPpathway)
pairwiseVIPpathway_allModules
```

```{r}
write.table(pairwiseVIPpathway_allModules, "metaboanalyst/all-module-pairwise-VIP-pathways.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Make heatmaps of each module's VIP

I'm only going to make heatmaps for the modules significantly correlated to a treatment x day condition.

##### Module 0

```{r}
#pdf("metaboanalyst/figures/module0_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module0$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module0_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module0$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module0_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module0$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 2

```{r}
#pdf("metaboanalyst/figures/module2_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module2$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module2_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module2$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module2_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module2$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 3

```{r}
#pdf("metaboanalyst/figures/module3_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module3$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module3_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module3$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module3_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module3$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 4

```{r}
#pdf("metaboanalyst/figures/module4_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module4$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module4_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module4$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module4_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module4$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 5

```{r}
#pdf("metaboanalyst/figures/module5_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module5$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module5_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module5$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module5_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module5$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 6

```{r}
#pdf("metaboanalyst/figures/module6_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module6$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module6_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module6$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module6_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module6$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 7

```{r}
#pdf("metaboanalyst/figures/module7_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module7$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module7_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module7$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module7_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module7$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

# ASCA

My WCNAs grouped compounds into module eigengenes and correlated those with temperature and physiological traits. I am going to use an ASCA to classify trends in metabolite abundance over temperature and time. Similar to the WCNA, this will provide me groups of compounds, but in a more statistically rigorous manner.

- [ALASCA tutorial](https://andjar.github.io/ALASCA/articles/ALASCA.html)
- [Ariana's lab notebook](https://ahuffmyer.github.io/ASH_Putnam_Lab_Notebook/Analysis-of-Variance-Simultaneous-Component-Analyses-ASCA-for-CEABIGR-project/)
- [Ariana's code from CEABiGR](https://github.com/sr320/ceabigr/blob/main/code/00-77-asca-exon.Rmd)

```{bash}
mkdir ASCA
mkdir ASCA/figures
```

## Format data

```{r}
metabolomics.ASCA <- rowwiseMetabData %>%
  log(.) %>%
  rownames_to_column(., var = "crab.ID") %>%
  pivot_longer(., cols = !crab.ID, names_to = "variable", values_to = "value") %>%
  left_join(metabolomicsMetadata, ., by = "crab.ID") %>%
  dplyr::select(-treatment_day) %>%
  mutate(., day = str_trim(day, side = "both")) %>%
  mutate(day = case_when(day == "3" ~ "04",
                         day != "3" ~ day)) %>%
  mutate(day = as.factor(day)) #Log-transform data. Pivot data longer for ASCA. Compound information should be "variable." Add metadata to dataframe with all compound information. Update metadata for day and reconvert day to a factor
head(metabolomics.ASCA) #Confirm changes
```

## Run model

```{r}
metabolomics.ASCAmodel <- ALASCA(df = metabolomics.ASCA,
                                 formula = value ~ day*treatment + (1|crab.ID),
                                 n_validation_runs = 100,
                                 validate = TRUE,
                                 reduce_dimensions = TRUE) #Run ASCA to identify day and temperature effects on compound abundance
```

## Interpret ASCA

```{r}
# pdf("ASCA/figures/scree-plot.pdf")
plot(metabolomics.ASCAmodel, component = 1:6, type = "scree") #Scree plot of variance explained by each PC. There were 6 components identified, with the sixth component explaining very little variance
# dev.off()
```

I'm now going to investigate the trends of the first five components, since those were the ones that explained any variance.

### Component 1

This component shows metabolites with consistently higher or lower abundance than 5ºC or 13ºC that change abundance (generally increasing) at 30ºC over time.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 1, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC1.pdf", height = 11, width = 8.5)
```
```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 1, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC1 <- get_loadings(metabolomics.ASCAmodel, component = 1, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC1)
```

```{r}
loadingsPC1.annot <- loadingsPC1[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC1.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC1.annot, "ASCA/loadings-PC1-annotations.csv", quote = FALSE, row.names = FALSE)
```

#### Metaboanalyst enrichment results

```{r}
topPathwayResultsPC1 <- read.csv("ASCA/PC1_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR)) %>%
  arrange(desc(enrichmentRatio)) %>%
  mutate(., order = rev(1:18)) #Import MetaboAnalyst results and modify for figure
head(topPathwayResultsPC1) #Confirm dataframe creation
```

```{r}
topPathwayResultsPC1 %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResultsPC1$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("ASCA/figures/PC1-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

### Component 2

This component highlights compounds in 5ºC that have similar abundance to 13ºC in day 4, but different in day 22.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 2, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 2, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC2 <- get_loadings(metabolomics.ASCAmodel, component = 2, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC2)
```

```{r}
loadingsPC2.annot <- loadingsPC2[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC2.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC2.annot, "ASCA/loadings-PC2-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Component 3

Absolute abundance in 5ºC changes over time but 30ºC remains stable

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 3, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 3, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC3 <- get_loadings(metabolomics.ASCAmodel, component = 3, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC3)
```

```{r}
loadingsPC3.annot <- loadingsPC3[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC3.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC3.annot, "ASCA/loadings-PC3-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Component 4

Absolute change in abundance over time in 30ºC is more extreme than at 5ºC.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 4, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC4.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 4, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC4.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC4 <- get_loadings(metabolomics.ASCAmodel, component = 4, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC4)
```

```{r}
loadingsPC4.annot <- loadingsPC4[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC4.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC4.annot, "ASCA/loadings-PC4-annotations.csv", quote = FALSE, row.names = FALSE)
```

#### Metaboanalyst enrichment results

```{r}
topPathwayResultsPC4 <- read.csv("ASCA/PC4_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR)) %>%
  arrange(desc(enrichmentRatio)) %>%
  mutate(., order = rev(1:24)) #Import MetaboAnalyst results and modify for figure
head(topPathwayResultsPC4) #Confirm dataframe creation
```

```{r}
topPathwayResultsPC4 %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResultsPC4$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("ASCA/figures/PC4-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

### Component 5

Absolute trends at 13ºC vary, absolute trends at 5ºC and 30ºC are similar.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 5, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC5.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 5, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC5.pdf", height = 11, width = 8.5)
```

Component 5 explains < 10% of the variance in this dataset. Additionally, the top 10 compounds strongly loaded onto PC5 have errors that cross 0. I think this makes the PC5 results difficult to interpret. I will go ahead and say that PCs that explain > 10% variance are my cutoff for analysis.

It's still good to have the list of loadings, so I'll save those anyways.

```{r}
loadingsPC5 <- get_loadings(metabolomics.ASCAmodel, component = 5, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC5)
```

```{r}
loadingsPC5.annot <- loadingsPC5[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC5.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC5.annot, "ASCA/loadings-PC5-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Final figure

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = seq(1,4), type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-all-components.pdf", height = 11, width = 8.5)
```
